{"nbformat":4,"nbformat_minor":0,"metadata":{"kernelspec":{"display_name":"Python 3","language":"python","name":"python3"},"language_info":{"codemirror_mode":{"name":"ipython","version":3},"file_extension":".py","mimetype":"text/x-python","name":"python","nbconvert_exporter":"python","pygments_lexer":"ipython3","version":"3.8.3"},"colab":{"name":"Project 2 - Antibiotic Resistance","provenance":[{"file_id":"1gW1ZdAu5SLSky4lQGoecWsMRGpuzmK4h","timestamp":1623455564127}],"collapsed_sections":["Ks6IOjoTn8PO"]}},"cells":[{"cell_type":"code","metadata":{"id":"-O95NAqt56oH"},"source":["from google.colab import drive\n","drive.mount('/content/drive')"],"execution_count":null,"outputs":[]},{"cell_type":"code","metadata":{"id":"SnWGTKEJ4RhZ"},"source":["!pip install biopython\n","import numpy as np\n","import pandas as pd\n","import sklearn.model_selection\n","import matplotlib.pyplot as plt\n","from Bio import SeqIO\n","from Bio.SeqIO.FastaIO import SimpleFastaParser"],"execution_count":null,"outputs":[]},{"cell_type":"markdown","metadata":{"id":"9CpE4EBLnYEo"},"source":["# FASTA to DataFrame of Sequences"]},{"cell_type":"code","metadata":{"id":"4-tQJaUPSHna"},"source":["def fastaDF (input_file,gene_n,drug):\n","  \"\"\"converts a FASTA file to a pandas DataFrame. \n","    parameters:\n","    input_file is the file path from current directory to the file. should be a .fa or .fasta file (string)\n","    gene_n is the name of the gene for which the sequences are in the FASTA file (string)\n","    drug : name of the antibiotic that the species have resistance for (string)\n","  \"\"\"\n","\n","  #import cipro gyrA FASTA and make DataFrame\n","  with open(input_file) as fasta_file: #cipro\n","      identifiers = []\n","      bacteria_IDs = []\n","      names = []\n","      lengths = []\n","      sequences = []\n","      gene = []\n","      for seq_record in SeqIO.parse(fasta_file, 'fasta'):  # (generator)\n","          identifiers.append(seq_record.id)\n","          lengths.append(len(seq_record.seq))\n","          s,d = seq_record.description.split('[')\n","          d = d[0:len(d)-1]\n","          bacteria_IDs.append(d)\n","          bact_name = d.split(' ')[0:2]\n","          n = ' '.join(bact_name[0:2])\n","          names.append(n)\n","          sequences.append(seq_record.seq)\n","      gene.append(names)\n","      gene.append(bacteria_IDs)\n","      gene.append(sequences)\n","      gene_df = pd.DataFrame(gene,index=['Bacteria Name','Strain',f'{gene_n} Sequence {drug}']).transpose()\n","  df1=gene_df.set_index('Bacteria Name')\n","  return df1"],"execution_count":null,"outputs":[]},{"cell_type":"code","metadata":{"id":"uzi07hegVUyb"},"source":["#Cipro gyrA FASTA to DF\n","gyr_path = '/content/drive/Shareddrives/Project2_Drive/PROJECT 2/combined_sequences/combined_gyra.fasta'\n","df_gyrA_cipro=fastaDF(gyr_path,'gyrA','C')\n","df_parC_cipro=fastaDF('/content/drive/Shareddrives/Project2_Drive/PROJECT 2/combined_sequences/combined_parc.fasta','parC','C')\n","df_gyrA_moxi = fastaDF('/content/drive/Shareddrives/Project2_Drive/PROJECT 2/combined_sequences/gyrA_moxi.fasta','gyrA','M')\n","df_parC_moxi = fastaDF('/content/drive/Shareddrives/Project2_Drive/PROJECT 2/combined_sequences/parC_moxi.fasta', 'parC', 'M')"],"execution_count":null,"outputs":[]},{"cell_type":"code","metadata":{"jupyter":{"outputs_hidden":true,"source_hidden":true},"id":"eB4y4ZwC4Rhg"},"source":["df_cipro = pd.merge(df_gyrA_cipro,df_parC_cipro,how=\"inner\", on=['Bacteria Name','Strain'])\n","df_cipro = df_cipro.drop(\"Strain\",axis=1)\n","df_cipro.sort_index()\n","df_cipro"],"execution_count":null,"outputs":[]},{"cell_type":"code","metadata":{"id":"ahf15ytx7tMN"},"source":["df_moxi = pd.merge(df_gyrA_moxi,df_parC_moxi,how=\"inner\", on=['Bacteria Name'],suffixes=(': gyrA moxi',': parC moxi'))\n","df_moxi = df_moxi.sort_index()\n","df_moxi=df_moxi.drop_duplicates(\"Strain: gyrA moxi\",keep='first')\n","df_moxi=df_moxi.drop_duplicates(\"Strain: parC moxi\",keep='first')\n","df_moxi=df_moxi.drop([\"Strain: parC moxi\",'Strain: gyrA moxi'],axis=1)\n","df_moxi.sort_index()\n","df_moxi"],"execution_count":null,"outputs":[]},{"cell_type":"code","metadata":{"id":"GgiBSp8jKNiB"},"source":[""],"execution_count":null,"outputs":[]},{"cell_type":"code","metadata":{"id":"E5JIEGR_730A"},"source":["df = pd.concat([df_moxi,df_cipro],join='outer')\n","df"],"execution_count":null,"outputs":[]},{"cell_type":"markdown","metadata":{"id":"MVXomTe5nlk5"},"source":["# Bag of Words (K-mers)"]},{"cell_type":"code","metadata":{"id":"rVlBug1YaX09"},"source":["def makeKmers(sequ,gene,drug):\n","  '''\n","  create list of k-mer words in DataFrame, along with bacteria name as index\n","  gene is string name\n","  drug is single letter abbreviation of drug name\n","  '''\n","\n","  def getKmers(sequence,ksize=6):\n","    '''\n","    function to convert sequence strings into k-mer words, default size = 6 (hexamer words)\n","    sequence input must be in String format\n","    '''\n","    return [sequence[x:x+ksize] for x in range(len(sequence) - ksize + 1)]\n","\n","  ind1 = []\n","  seq1 = []\n","  df_test = []\n","  for i, s in sequ.iterrows():\n","      ind1.append(i)\n","      seq1.append(str(s[f'{gene} Sequence {drug}']))\n","  df_test.append(ind1)\n","  df_test.append(seq1)\n","  df_test = pd.DataFrame(df_test,index=['Bacteria Name',f'{gene}-{drug} Seq']).transpose()\n","  df_test=df_test.set_index(['Bacteria Name'])\n","  df_test=df_test.sort_index()\n","  df=df_test\n","  df[f'{gene}-{drug}']=df.apply(lambda x: getKmers(x[f'{gene}-{drug} Seq']), axis=1)\n","  df = df.drop([f'{gene}-{drug} Seq'],axis=1)\n","  return df"],"execution_count":null,"outputs":[]},{"cell_type":"code","metadata":{"id":"myf2MwSxd0cF"},"source":["#k-mer DF for cipro genes\n","df_1 = makeKmers(df_cipro,'gyrA','C')\n","df_2 = makeKmers(df_cipro,'parC','C')\n","df_words = df_1.merge(df_2,on='Bacteria Name')\n","df_words"],"execution_count":null,"outputs":[]},{"cell_type":"code","metadata":{"id":"Q22JzWHlh3zR"},"source":["#k-mer DF for cipro genes\n","df_1 = makeKmers(df_moxi,'gyrA','M')\n","df_2 = makeKmers(df_moxi,'parC','M')\n","df_words_m = df_1.merge(df_2,on='Bacteria Name')\n","df_words_m"],"execution_count":null,"outputs":[]},{"cell_type":"code","metadata":{"id":"LtrtWCSyIND_"},"source":["df_words_m.to_csv('/content/drive/Shareddrives/Project2_Drive/PROJECT 2/df_m.csv')"],"execution_count":null,"outputs":[]},{"cell_type":"code","metadata":{"id":"OrhwRqlyx4cW"},"source":["gyrA_C_word = list(df_words['gyrA-C'])\n","for item in range(len(gyrA_C_word)):\n","    gyrA_C_word[item] = ' '.join(gyrA_C_word[item])\n","C_spec = df_words.index.values                         #C_ for C species\n","parC_C_word = list(df_words['parC-C'])\n","for item in range(len(parC_C_word)):\n","    parC_C_word[item] = ' '.join(parC_C_word[item])"],"execution_count":null,"outputs":[]},{"cell_type":"code","metadata":{"id":"3cMyo9bZ0D1E"},"source":["gyrA_M_word = list(df_words_m['gyrA-M'])\n","for item in range(len(gyrA_M_word)):\n","    gyrA_M_word[item] = ' '.join(gyrA_M_word[item])\n","M_spec = df_words_m.index.values                         #M_ for moxi species\n","parC_M_word = list(df_words_m['parC-M'])\n","for item in range(len(parC_M_word)):\n","    parC_M_word[item] = ' '.join(parC_M_word[item])"],"execution_count":null,"outputs":[]},{"cell_type":"code","metadata":{"id":"27a83xGlxcZ-"},"source":["spec=np.append(C_spec,M_spec)\n","gyrA_words=gyrA_C_word+gyrA_M_word\n","parC_words=parC_C_word+parC_M_word"],"execution_count":null,"outputs":[]},{"cell_type":"markdown","metadata":{"id":"hHiNsWfuta34"},"source":[""]},{"cell_type":"code","metadata":{"id":"7AxEzcyn0upT"},"source":["'''\n"," Creating the Bag of Words model using CountVectorizer()\n","# This is equivalent to k-mer counting\n","# The n-gram size of 4 was previously determined by testing\n","from sklearn.feature_extraction.text import CountVectorizer\n","cv = CountVectorizer(ngram_range=(4,4))\n","gAC = cv.fit_transform(human_texts)\n","pCC = cv.transform(chimp_texts)\n","gAM = cv.transform(dog_texts)\n","'''"],"execution_count":null,"outputs":[]},{"cell_type":"code","metadata":{"id":"7ttWY_5awWXV"},"source":["df_kmer = pd.concat([df_words,df_words_m],join='outer')\n","df_kmer"],"execution_count":null,"outputs":[]},{"cell_type":"code","metadata":{"id":"1p2-62zpryE2"},"source":["from sklearn.feature_extraction.text import CountVectorizer\n","cv = CountVectorizer(ngram_range=(4,4)) #The n-gram size of 4 is previously determined by testing\n","X_gyrA = cv.fit_transform(gyrA_words)\n","X_parC = cv.transform(parC_words)\n","print(X_gyrA.shape)\n","print(X_parC.shape)"],"execution_count":null,"outputs":[]},{"cell_type":"markdown","metadata":{"id":"hl2FQdbwnv9-"},"source":["# MIC DataFrame (Y values)"]},{"cell_type":"code","metadata":{"id":"CmvlLuY54Rhh"},"source":["weighted_MIC = pd.read_csv(\"/content/drive/Shareddrives/Project2_Drive/PROJECT 2/MIC_resistance_all.csv\")\n","weighted_MIC = weighted_MIC.set_index(\"Bacteria Name\")\n","weighted_MIC.dropna(axis=0,how=\"all\",inplace=True)\n","weighted_MIC=weighted_MIC.drop('Staphylococcus hominis',axis=0)\n","weighted_MIC"],"execution_count":null,"outputs":[]},{"cell_type":"code","metadata":{"id":"e_Id14TF4Rhi"},"source":["MIC_df_cipro = weighted_MIC[[\"Ciprofloxacin Weighted MIC\"]]\n","MIC_df_cipro = MIC_df_cipro.dropna(how='any')\n","MIC_df_cipro = MIC_df_cipro.rename(columns={\"Ciprofloxacin Weighted MIC\":\"cipro MIC\"})\n","MIC_df_cipro\n","y_cipro = MIC_df_cipro.iloc[:, 0].values # y_cipro for Cipro MIC\n","len(y_cipro)"],"execution_count":null,"outputs":[]},{"cell_type":"code","metadata":{"id":"iYq9Co45ASEP"},"source":["MIC_df_moxi = weighted_MIC[[\"Moxifloxacin Weighted MIC\"]]\n","MIC_df_moxi = MIC_df_moxi.dropna(how='any')\n","MIC_df_moxi = MIC_df_moxi.rename(columns={\"Moxifloxacin Weighted MIC\":\"moxi MIC\"})\n","len(MIC_df_moxi)\n","#y_moxi = MIC_df_moxi.iloc[:, 0].values # y_moxi for moxi MIC\n","#y_moxi"],"execution_count":null,"outputs":[]},{"cell_type":"code","metadata":{"id":"OEuA_J1_IE71"},"source":[""],"execution_count":null,"outputs":[]},{"cell_type":"code","metadata":{"id":"CHYy_q83zAI5"},"source":["y_MIC=np.append(y_cipro,y_moxi)"],"execution_count":null,"outputs":[]},{"cell_type":"code","metadata":{"id":"NS_jz_dJIcSH"},"source":["MIC_df = pd.concat([MIC_df_cipro,MIC_df_moxi],join='outer')\n","MIC_df"],"execution_count":null,"outputs":[]},{"cell_type":"markdown","metadata":{"id":"Ks6IOjoTn8PO"},"source":["# Percent Identity values"]},{"cell_type":"code","metadata":{"id":"mRQBwXVB4Rhi"},"source":["PIM_df_cipro = pd.read_csv(\"/content/drive/Shareddrives/Project2_Drive/PROJECT 2/ciprofloaxin_PIM_singleval.csv\")\n","PIM_df_cipro = PIM_df_cipro.set_index(\"Bacteria Name\")\n","PIM_df_cipro = PIM_df_cipro.sort_index()\n","PIM_df_cipro = PIM_df_cipro.rename(columns={\"gyrA\":\"c-gyrA\", \"parC\": \"c-parC\"})\n","PIM_df_cipro"],"execution_count":null,"outputs":[]},{"cell_type":"code","metadata":{"id":"zjtie9-BDnad"},"source":["PIM_df_moxi = pd.read_csv(\"/content/drive/Shareddrives/Project2_Drive/PROJECT 2/moxi_PIM.csv\")\n","PIM_df_moxi = PIM_df_moxi.set_index(\"Bacteria Name\")\n","PIM_df_moxi = PIM_df_moxi.sort_index()\n","PIM_df_moxi = PIM_df_moxi.rename(columns={\"gyrA PIM\":\"m-gyrA\", \"parC PIM\": \"m-parC\"})\n","PIM_df_moxi"],"execution_count":null,"outputs":[]},{"cell_type":"code","metadata":{"id":"LIp2V-SIIJib"},"source":["PIM_df=pd.concat([PIM_df_cipro,PIM_df_moxi],join='outer',sort=True)\n","PIM_df.groupby(level=0).sum()\n","PIM_df"],"execution_count":null,"outputs":[]},{"cell_type":"markdown","metadata":{"id":"zSnBHNgxoF0m"},"source":["# Combining all data into DF"]},{"cell_type":"code","metadata":{"id":"VJiSnrBg4Rhj"},"source":["cipro_df = pd.merge(MIC_df_cipro,PIM_df_cipro,how=\"inner\",on=[\"Bacteria Name\"])\n","cipro_df.drop_duplicates()\n","cipro_df2=pd.merge(cipro_df,df_words,how=\"inner\",on=[\"Bacteria Name\"])\n","cipro_df2.drop_duplicates(subset='cipro MIC')\n","cipro_df2 #ALL CIPRO DATA"],"execution_count":null,"outputs":[]},{"cell_type":"code","metadata":{"id":"ZRZxTBPbEq83"},"source":["moxi_df = pd.merge(MIC_df_moxi,PIM_df_moxi,how=\"inner\",on=[\"Bacteria Name\"])\n","moxi_df.groupby(level=0)\n","moxi_df=moxi_df.drop_duplicates()\n","moxi_df"],"execution_count":null,"outputs":[]},{"cell_type":"code","metadata":{"id":"gRxZxEYcOI3S"},"source":["moxi_df2 = pd.merge(moxi_df,df_words_m,how=\"inner\",on=[\"Bacteria Name\"])\n","moxi_df2.drop_duplicates(subset='moxi MIC') #ALL MOXI DATA"],"execution_count":null,"outputs":[]},{"cell_type":"markdown","metadata":{"id":"SA29n2u39qkw"},"source":["\n","```\n","# df = pd.merge(df,MIC_df_cipro,how=\"inner\", on=['Bacteria Name'])\n","df\n","gyrA_PIM = pd.read_csv(\"/content/drive/Shareddrives/Project2_Drive/Project 1/gyrA_PIM.csv\")\n","gyrA_PIM = gyrA_PIM.set_index(\"Bacteria Name\")\n","gyrA_PIM\n","df2 = pd.merge(gyrA_PIM,df,how=\"inner\", on=['Bacteria Name','Strain'])\n","df2\n","df_rf = df2.drop([\"Strain\",\"Sequence: gyrA\",\"Sequence: parC\"],axis=1)\n","df_rf\n","```\n","\n"]},{"cell_type":"markdown","metadata":{"id":"SVGr5kdsoUJc"},"source":["# Data Separation for ML"]},{"cell_type":"code","metadata":{"collapsed":true,"jupyter":{"outputs_hidden":true},"id":"wDXkmNLS4Rhp"},"source":["MIC_val_cipro = np.array(cipro_df['cipro MIC'])\n","columns_list = list(cipro_df.columns)\n","species_list = list(cipro_df.index)\n","cipro_df = cipro_df.drop(\"cipro MIC\", axis=1)\n","cipro_df"],"execution_count":null,"outputs":[]},{"cell_type":"code","metadata":{"id":"p3ydFv2nQ0QQ"},"source":["cipro_df = np.array(cipro_df)\n","cipro_df"],"execution_count":null,"outputs":[]},{"cell_type":"code","metadata":{"id":"_cebp_J7vEuj"},"source":[""],"execution_count":null,"outputs":[]},{"cell_type":"markdown","metadata":{"id":"JS50MSIwoZSR"},"source":["# Random Forest Regression K-mer"]},{"cell_type":"code","metadata":{"id":"kMi9-NzR4Rhr"},"source":["train_features, test_features, train_labels, test_labels = sklearn.model_selection.train_test_split(cipro_df, MIC_val_cipro, test_size = 0.4,random_state=38,shuffle=False)"],"execution_count":null,"outputs":[]},{"cell_type":"code","metadata":{"id":"BSKDx0E4Sj2l"},"source":["test_labels"],"execution_count":null,"outputs":[]},{"cell_type":"code","metadata":{"id":"ZpElVU-P4Rhs"},"source":["print('Training Features Shape:', train_features.shape)\n","print('Training Labels Shape:', train_labels.shape)\n","print('Testing Features Shape:', test_features.shape)\n","print('Testing Labels Shape:', test_labels.shape)"],"execution_count":null,"outputs":[]},{"cell_type":"code","metadata":{"id":"kCMe3P8x4Rht"},"source":["from sklearn.ensemble import RandomForestRegressor\n","# Instantiate model with 1000 decision trees\n","rf = RandomForestRegressor(n_estimators = 1000, random_state = 18)\n","# Train the model on training data\n","rf.fit(train_features, train_labels)"],"execution_count":null,"outputs":[]},{"cell_type":"code","metadata":{"id":"V4NoP_7n4Rhu"},"source":["predictions = rf.predict(test_features)\n","# Calculate the absolute errors\n","errors = abs(predictions - test_labels)\n","print(predictions)\n","print(test_labels)\n","# Print out the mean absolute error (mae)\n","print('Mean Absolute Error:', round(np.mean(errors), 2))"],"execution_count":null,"outputs":[]},{"cell_type":"code","metadata":{"id":"yLBgEti94Rhx"},"source":["# Calculate mean absolute percentage error (MAPE)\n","mape = 100 * (errors / test_labels)\n","# Calculate and display accuracy\n","accuracy = 100 - np.mean(mape)\n","print('Accuracy:', round(accuracy, 2), '%.')"],"execution_count":null,"outputs":[]},{"cell_type":"markdown","metadata":{"id":"pAZ1lSDa1G29"},"source":["# Model Comparison"]},{"cell_type":"code","metadata":{"id":"lJS-0XTe4Rhz"},"source":["# Imports\n","import plotly.graph_objects as go\n","\n","# Grab the data\n","bacteria_names = cipro_df2.index.array\n","model1_predictions = predictions\n","model2_predictions = rf.predict(test_features)\n","actual_values = test_labels\n","\n","# Testing\n","print(len(bacteria_names))\n","print(len(model1_predictions))\n","print(len(model2_predictions))\n","print(len(actual_values))\n","\n","# Plot the chart\n","fig = go.Figure(\n","  data=[\n","    go.Bar(name='Model 1 Predicted MIC', x=bacteria_names, y=model1_predictions),\n","    go.Bar(name='Model 2 Predicted MIC', x=bacteria_names, y=model2_predictions),\n","    go.Bar(name='Actual MIC', x=bacteria_names, y=actual_values)\n","  ],\n","  layout={\n","    'yaxis': {'title': 'Predicted MIC Value'},\n","    'xaxis': {'title': 'Bacteria Name'},\n","    'title': 'Bacteria Names vs. Predicted MIC Value'\n","  }\n",")\n","fig.update_layout(barmode='group', xaxis_tickangle=-45)\n","fig.show()"],"execution_count":null,"outputs":[]}]}