{"nbformat":4,"nbformat_minor":0,"metadata":{"kernelspec":{"display_name":"Python 3","language":"python","name":"python3"},"language_info":{"codemirror_mode":{"name":"ipython","version":3},"file_extension":".py","mimetype":"text/x-python","name":"python","nbconvert_exporter":"python","pygments_lexer":"ipython3","version":"3.8.3"},"colab":{"name":"Project 2 - Predicting Antibiotic Resistance.ipynb","provenance":[],"collapsed_sections":[]}},"cells":[{"cell_type":"markdown","metadata":{"id":"1qgcmoObgAJd"},"source":["# Project Setup\n","\n","You must have access to the \"PROJECT 2\" folder on Google Drive for this notebook to run properly. The notebook must also be run in Google Colab.\n","\n","https://drive.google.com/drive/folders/1VdeCGZHseDDQfzhV2jRnSmbxW39bp2_g?usp=sharing\n"]},{"cell_type":"markdown","metadata":{"id":"EqnaYBJLW70u"},"source":["## Mounting Google Drive\n","\n"]},{"cell_type":"code","metadata":{"id":"-O95NAqt56oH","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1623481806945,"user_tz":420,"elapsed":174032,"user":{"displayName":"Lila Balakrishnan","photoUrl":"","userId":"05882512769322768995"}},"outputId":"154de99c-bc63-4f77-d396-cf162f763a4f"},"source":["from google.colab import drive\n","drive.mount('/content/drive')"],"execution_count":2,"outputs":[{"output_type":"stream","text":["Mounted at /content/drive\n"],"name":"stdout"}]},{"cell_type":"markdown","metadata":{"id":"PZnuoxMwXXgu"},"source":["## Imports"]},{"cell_type":"code","metadata":{"id":"SnWGTKEJ4RhZ","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1623481812847,"user_tz":420,"elapsed":6064,"user":{"displayName":"Lila Balakrishnan","photoUrl":"","userId":"05882512769322768995"}},"outputId":"a558034a-e6c1-4118-d460-bdcf7ca19650"},"source":["!pip install biopython\n","import numpy as np\n","import pandas as pd\n","import sklearn.model_selection\n","import matplotlib.pyplot as plt\n","from Bio import SeqIO\n","from Bio.SeqIO.FastaIO import SimpleFastaParser"],"execution_count":3,"outputs":[{"output_type":"stream","text":["Collecting biopython\n","\u001b[?25l  Downloading https://files.pythonhosted.org/packages/5a/42/de1ed545df624180b84c613e5e4de4848f72989ce5846a74af6baa0737b9/biopython-1.79-cp37-cp37m-manylinux_2_5_x86_64.manylinux1_x86_64.whl (2.3MB)\n","\u001b[K     |████████████████████████████████| 2.3MB 6.3MB/s \n","\u001b[?25hRequirement already satisfied: numpy in /usr/local/lib/python3.7/dist-packages (from biopython) (1.19.5)\n","Installing collected packages: biopython\n","Successfully installed biopython-1.79\n"],"name":"stdout"}]},{"cell_type":"markdown","metadata":{"id":"9CpE4EBLnYEo"},"source":["# Helper Functions"]},{"cell_type":"markdown","metadata":{"id":"KLk36p4GXb2e"},"source":["## FASTA to DataFrame of Sequences"]},{"cell_type":"code","metadata":{"id":"4-tQJaUPSHna","executionInfo":{"status":"ok","timestamp":1623481812850,"user_tz":420,"elapsed":22,"user":{"displayName":"Lila Balakrishnan","photoUrl":"","userId":"05882512769322768995"}}},"source":["def fastaDF (input_file,gene_n,drug):\n","  \"\"\"converts a FASTA file to a pandas DataFrame. \n","    parameters:\n","    input_file is the file path from current directory to the file. should be a .fa or .fasta file (string)\n","    gene_n is the name of the gene for which the sequences are in the FASTA file (string)\n","    drug : name of the antibiotic that the species have resistance for (string)\n","  \"\"\"\n","\n","  #import cipro gyrA FASTA and make DataFrame\n","  with open(input_file) as fasta_file: #cipro\n","      identifiers = []\n","      bacteria_IDs = []\n","      names = []\n","      lengths = []\n","      sequences = []\n","      gene = []\n","      for seq_record in SeqIO.parse(fasta_file, 'fasta'):  # (generator)\n","          identifiers.append(seq_record.id)\n","          lengths.append(len(seq_record.seq))\n","          s,d = seq_record.description.split('[')\n","          d = d[0:len(d)-1]\n","          bacteria_IDs.append(d)\n","          bact_name = d.split(' ')[0:2]\n","          n = ' '.join(bact_name[0:2])\n","          names.append(n)\n","          sequences.append(seq_record.seq)\n","      gene.append(names)\n","      gene.append(bacteria_IDs)\n","      gene.append(sequences)\n","      gene_df = pd.DataFrame(gene,index=['Bacteria Name','Strain',f'{gene_n} Sequence {drug}']).transpose()\n","  df1=gene_df.set_index('Bacteria Name')\n","  return df1"],"execution_count":4,"outputs":[]},{"cell_type":"markdown","metadata":{"id":"MVXomTe5nlk5"},"source":["## Bag of Words (K-mers)"]},{"cell_type":"code","metadata":{"id":"rVlBug1YaX09","executionInfo":{"status":"ok","timestamp":1623481812851,"user_tz":420,"elapsed":18,"user":{"displayName":"Lila Balakrishnan","photoUrl":"","userId":"05882512769322768995"}}},"source":["def makeKmers(sequ,gene,drug):\n","  '''\n","  create list of k-mer words in DataFrame, along with bacteria name as index\n","  gene is string name\n","  drug is single letter abbreviation of drug name\n","  '''\n","\n","  def getKmers(sequence,ksize=6):\n","    '''\n","    function to convert sequence strings into k-mer words, default size = 6 (hexamer words)\n","    sequence input must be in String format\n","    '''\n","    return [sequence[x:x+ksize] for x in range(len(sequence) - ksize + 1)]\n","\n","  ind1 = []\n","  seq1 = []\n","  df_test = []\n","  for i, s in sequ.iterrows():\n","      ind1.append(i)\n","      seq1.append(str(s[f'{gene} Sequence {drug}']))\n","  df_test.append(ind1)\n","  df_test.append(seq1)\n","  df_test = pd.DataFrame(df_test,index=['Bacteria Name',f'{gene}-{drug} Seq']).transpose()\n","  df_test=df_test.set_index(['Bacteria Name'])\n","  df_test=df_test.sort_index()\n","  df=df_test\n","  df[f'{gene}-{drug}']=df.apply(lambda x: getKmers(x[f'{gene}-{drug} Seq']), axis=1)\n","  df = df.drop([f'{gene}-{drug} Seq'],axis=1)\n","  return df"],"execution_count":5,"outputs":[]},{"cell_type":"markdown","metadata":{"id":"vKKSLptCXhHF"},"source":["# Data Processing"]},{"cell_type":"code","metadata":{"id":"uzi07hegVUyb","executionInfo":{"status":"ok","timestamp":1623481815207,"user_tz":420,"elapsed":2371,"user":{"displayName":"Lila Balakrishnan","photoUrl":"","userId":"05882512769322768995"}}},"source":["#Cipro gyrA FASTA to DF\n","gyr_path = '/content/drive/Shareddrives/Project2_Drive/PROJECT 2/combined_sequences/combined_gyra.fasta'\n","df_gyrA_cipro=fastaDF(gyr_path,'gyrA','C')\n","df_parC_cipro=fastaDF('/content/drive/Shareddrives/Project2_Drive/PROJECT 2/combined_sequences/combined_parc.fasta','parC','C')\n","df_gyrA_moxi = fastaDF('/content/drive/Shareddrives/Project2_Drive/PROJECT 2/combined_sequences/gyrA_moxi.fasta','gyrA','M')\n","df_parC_moxi = fastaDF('/content/drive/Shareddrives/Project2_Drive/PROJECT 2/combined_sequences/parC_moxi.fasta', 'parC', 'M')"],"execution_count":6,"outputs":[]},{"cell_type":"code","metadata":{"jupyter":{"outputs_hidden":true,"source_hidden":true},"id":"eB4y4ZwC4Rhg","colab":{"base_uri":"https://localhost:8080/","height":917},"executionInfo":{"status":"ok","timestamp":1623481815209,"user_tz":420,"elapsed":214,"user":{"displayName":"Lila Balakrishnan","photoUrl":"","userId":"05882512769322768995"}},"outputId":"72a3cbae-2038-4843-84a0-72e85bad23ee"},"source":["df_cipro = pd.merge(df_gyrA_cipro,df_parC_cipro,how=\"inner\", on=['Bacteria Name','Strain'])\n","df_cipro = df_cipro.drop(\"Strain\",axis=1)\n","df_cipro = df_cipro.drop(\"Staphylococcus hominis\",axis=0)\n","df_cipro.sort_index()"],"execution_count":7,"outputs":[{"output_type":"execute_result","data":{"text/html":["<div>\n","<style scoped>\n","    .dataframe tbody tr th:only-of-type {\n","        vertical-align: middle;\n","    }\n","\n","    .dataframe tbody tr th {\n","        vertical-align: top;\n","    }\n","\n","    .dataframe thead th {\n","        text-align: right;\n","    }\n","</style>\n","<table border=\"1\" class=\"dataframe\">\n","  <thead>\n","    <tr style=\"text-align: right;\">\n","      <th></th>\n","      <th>gyrA Sequence C</th>\n","      <th>parC Sequence C</th>\n","    </tr>\n","    <tr>\n","      <th>Bacteria Name</th>\n","      <th></th>\n","      <th></th>\n","    </tr>\n","  </thead>\n","  <tbody>\n","    <tr>\n","      <th>Acinetobacter lwoffii</th>\n","      <td>(a, t, g, a, g, c, g, t, a, t, c, g, g, a, a, ...</td>\n","      <td>(a, t, g, a, c, a, a, g, c, c, t, t, g, c, g, ...</td>\n","    </tr>\n","    <tr>\n","      <th>Aerococcus urinae</th>\n","      <td>(a, t, g, g, t, t, g, a, a, g, a, a, c, a, t, ...</td>\n","      <td>(a, t, g, g, c, g, a, t, t, g, a, t, a, t, t, ...</td>\n","    </tr>\n","    <tr>\n","      <th>Alcaligenes faecalis</th>\n","      <td>(a, t, g, g, a, t, t, c, c, t, t, t, g, c, c, ...</td>\n","      <td>(a, t, g, g, a, c, a, g, c, a, a, t, c, a, a, ...</td>\n","    </tr>\n","    <tr>\n","      <th>Citrobacter braakii</th>\n","      <td>(a, t, g, a, g, c, g, a, c, c, t, t, g, c, g, ...</td>\n","      <td>(a, t, g, a, g, c, g, a, t, a, t, g, g, c, a, ...</td>\n","    </tr>\n","    <tr>\n","      <th>Citrobacter freundii</th>\n","      <td>(a, t, g, a, g, c, g, a, c, c, t, t, g, c, g, ...</td>\n","      <td>(a, t, g, a, g, c, g, a, t, a, t, g, g, c, a, ...</td>\n","    </tr>\n","    <tr>\n","      <th>Clostridium perfringens</th>\n","      <td>(a, t, g, g, c, t, a, a, g, a, a, g, a, a, t, ...</td>\n","      <td>(a, t, g, a, g, c, t, t, a, a, a, t, g, a, g, ...</td>\n","    </tr>\n","    <tr>\n","      <th>Enterobacter cloacae</th>\n","      <td>(a, t, g, a, g, c, g, a, c, c, t, t, g, c, g, ...</td>\n","      <td>(a, t, g, a, g, c, g, a, t, a, t, g, g, c, a, ...</td>\n","    </tr>\n","    <tr>\n","      <th>Enterococcus faecalis</th>\n","      <td>(a, t, g, a, g, t, g, a, a, g, a, a, a, t, t, ...</td>\n","      <td>(t, t, g, g, a, a, a, a, a, c, g, c, c, a, a, ...</td>\n","    </tr>\n","    <tr>\n","      <th>Haemophilus influenzae</th>\n","      <td>(a, t, g, a, c, a, a, a, t, a, t, c, a, a, c, ...</td>\n","      <td>(a, t, g, a, c, g, g, a, t, t, c, a, a, t, c, ...</td>\n","    </tr>\n","    <tr>\n","      <th>Haemophilus parainfluenzae</th>\n","      <td>(a, t, g, a, c, g, g, a, t, t, c, a, a, t, c, ...</td>\n","      <td>(a, t, g, a, g, c, a, a, t, a, t, t, a, a, c, ...</td>\n","    </tr>\n","    <tr>\n","      <th>Klebsiella oxytoca</th>\n","      <td>(a, t, g, a, g, c, g, a, c, c, t, t, g, c, g, ...</td>\n","      <td>(a, t, g, a, g, c, g, a, t, a, t, g, g, c, a, ...</td>\n","    </tr>\n","    <tr>\n","      <th>Klebsiella pneumoniae</th>\n","      <td>(a, t, g, a, g, c, g, a, c, c, t, t, g, c, g, ...</td>\n","      <td>(a, t, g, a, g, c, g, a, t, a, t, g, g, c, a, ...</td>\n","    </tr>\n","    <tr>\n","      <th>Moraxella catarrhalis</th>\n","      <td>(a, t, g, a, g, t, g, a, t, t, c, g, g, t, c, ...</td>\n","      <td>(a, t, g, a, g, t, g, a, t, t, c, g, g, t, c, ...</td>\n","    </tr>\n","    <tr>\n","      <th>Morganella morganii</th>\n","      <td>(a, t, g, a, g, t, g, a, a, a, t, a, a, c, t, ...</td>\n","      <td>(a, t, g, a, g, t, g, a, a, a, t, a, a, c, t, ...</td>\n","    </tr>\n","    <tr>\n","      <th>Neisseria gonorrhoeae</th>\n","      <td>(a, t, g, a, c, c, g, a, c, g, c, a, a, c, c, ...</td>\n","      <td>(a, t, g, a, a, t, a, c, g, c, a, a, c, c, g, ...</td>\n","    </tr>\n","    <tr>\n","      <th>Neisseria meningitidis</th>\n","      <td>(a, t, g, a, c, c, g, a, c, g, c, a, a, c, c, ...</td>\n","      <td>(a, t, g, a, a, t, a, c, g, c, a, a, g, c, g, ...</td>\n","    </tr>\n","    <tr>\n","      <th>Serratia liquefaciens</th>\n","      <td>(a, t, g, a, g, c, g, a, c, c, t, t, g, c, c, ...</td>\n","      <td>(a, t, g, a, g, t, g, a, t, c, t, g, a, c, t, ...</td>\n","    </tr>\n","    <tr>\n","      <th>Serratia marcescens</th>\n","      <td>(a, t, g, a, g, c, g, a, c, c, t, t, g, c, c, ...</td>\n","      <td>(a, t, g, a, g, t, g, a, t, c, t, g, a, c, t, ...</td>\n","    </tr>\n","    <tr>\n","      <th>Shigella flexneri</th>\n","      <td>(a, t, g, a, g, c, g, a, c, c, t, t, g, c, g, ...</td>\n","      <td>(a, t, g, a, g, c, g, a, t, a, t, g, g, c, a, ...</td>\n","    </tr>\n","    <tr>\n","      <th>Shigella sonnei</th>\n","      <td>(a, t, g, a, g, c, g, a, c, c, t, t, g, c, g, ...</td>\n","      <td>(a, t, g, a, g, c, g, a, t, a, t, g, g, c, a, ...</td>\n","    </tr>\n","    <tr>\n","      <th>Staphylococcus aureus</th>\n","      <td>(a, t, g, g, c, t, g, a, a, t, t, a, c, c, t, ...</td>\n","      <td>(g, t, g, a, g, t, g, a, a, a, t, a, a, t, t, ...</td>\n","    </tr>\n","    <tr>\n","      <th>Staphylococcus haemolyticus</th>\n","      <td>(a, t, g, g, c, t, g, a, c, t, t, a, c, c, t, ...</td>\n","      <td>(t, t, g, a, g, t, g, a, g, a, t, a, a, t, t, ...</td>\n","    </tr>\n","    <tr>\n","      <th>Staphylococcus saprophyticus</th>\n","      <td>(a, t, g, g, c, t, g, a, a, t, t, a, c, c, t, ...</td>\n","      <td>(t, t, g, a, g, t, g, a, a, a, t, a, a, t, t, ...</td>\n","    </tr>\n","    <tr>\n","      <th>Streptococcus agalactiae</th>\n","      <td>(a, t, g, c, a, a, g, a, t, a, a, a, a, a, t, ...</td>\n","      <td>(a, t, g, t, c, t, a, a, t, a, t, t, c, a, a, ...</td>\n","    </tr>\n","    <tr>\n","      <th>Streptococcus mitis</th>\n","      <td>(a, t, g, c, a, g, g, a, t, a, g, a, a, a, t, ...</td>\n","      <td>(a, t, g, t, c, t, a, a, c, a, t, t, c, a, a, ...</td>\n","    </tr>\n","    <tr>\n","      <th>Streptococcus pyogenes</th>\n","      <td>(a, t, g, c, a, a, g, a, t, c, g, a, a, a, t, ...</td>\n","      <td>(a, t, g, t, c, a, a, a, c, a, t, t, c, a, a, ...</td>\n","    </tr>\n","    <tr>\n","      <th>Streptococcus viridans</th>\n","      <td>(a, t, g, c, a, g, g, a, t, a, a, t, a, a, c, ...</td>\n","      <td>(a, t, g, g, g, a, g, a, g, c, g, c, t, t, t, ...</td>\n","    </tr>\n","  </tbody>\n","</table>\n","</div>"],"text/plain":["                                                                gyrA Sequence C                                    parC Sequence C\n","Bacteria Name                                                                                                                     \n","Acinetobacter lwoffii         (a, t, g, a, g, c, g, t, a, t, c, g, g, a, a, ...  (a, t, g, a, c, a, a, g, c, c, t, t, g, c, g, ...\n","Aerococcus urinae             (a, t, g, g, t, t, g, a, a, g, a, a, c, a, t, ...  (a, t, g, g, c, g, a, t, t, g, a, t, a, t, t, ...\n","Alcaligenes faecalis          (a, t, g, g, a, t, t, c, c, t, t, t, g, c, c, ...  (a, t, g, g, a, c, a, g, c, a, a, t, c, a, a, ...\n","Citrobacter braakii           (a, t, g, a, g, c, g, a, c, c, t, t, g, c, g, ...  (a, t, g, a, g, c, g, a, t, a, t, g, g, c, a, ...\n","Citrobacter freundii          (a, t, g, a, g, c, g, a, c, c, t, t, g, c, g, ...  (a, t, g, a, g, c, g, a, t, a, t, g, g, c, a, ...\n","Clostridium perfringens       (a, t, g, g, c, t, a, a, g, a, a, g, a, a, t, ...  (a, t, g, a, g, c, t, t, a, a, a, t, g, a, g, ...\n","Enterobacter cloacae          (a, t, g, a, g, c, g, a, c, c, t, t, g, c, g, ...  (a, t, g, a, g, c, g, a, t, a, t, g, g, c, a, ...\n","Enterococcus faecalis         (a, t, g, a, g, t, g, a, a, g, a, a, a, t, t, ...  (t, t, g, g, a, a, a, a, a, c, g, c, c, a, a, ...\n","Haemophilus influenzae        (a, t, g, a, c, a, a, a, t, a, t, c, a, a, c, ...  (a, t, g, a, c, g, g, a, t, t, c, a, a, t, c, ...\n","Haemophilus parainfluenzae    (a, t, g, a, c, g, g, a, t, t, c, a, a, t, c, ...  (a, t, g, a, g, c, a, a, t, a, t, t, a, a, c, ...\n","Klebsiella oxytoca            (a, t, g, a, g, c, g, a, c, c, t, t, g, c, g, ...  (a, t, g, a, g, c, g, a, t, a, t, g, g, c, a, ...\n","Klebsiella pneumoniae         (a, t, g, a, g, c, g, a, c, c, t, t, g, c, g, ...  (a, t, g, a, g, c, g, a, t, a, t, g, g, c, a, ...\n","Moraxella catarrhalis         (a, t, g, a, g, t, g, a, t, t, c, g, g, t, c, ...  (a, t, g, a, g, t, g, a, t, t, c, g, g, t, c, ...\n","Morganella morganii           (a, t, g, a, g, t, g, a, a, a, t, a, a, c, t, ...  (a, t, g, a, g, t, g, a, a, a, t, a, a, c, t, ...\n","Neisseria gonorrhoeae         (a, t, g, a, c, c, g, a, c, g, c, a, a, c, c, ...  (a, t, g, a, a, t, a, c, g, c, a, a, c, c, g, ...\n","Neisseria meningitidis        (a, t, g, a, c, c, g, a, c, g, c, a, a, c, c, ...  (a, t, g, a, a, t, a, c, g, c, a, a, g, c, g, ...\n","Serratia liquefaciens         (a, t, g, a, g, c, g, a, c, c, t, t, g, c, c, ...  (a, t, g, a, g, t, g, a, t, c, t, g, a, c, t, ...\n","Serratia marcescens           (a, t, g, a, g, c, g, a, c, c, t, t, g, c, c, ...  (a, t, g, a, g, t, g, a, t, c, t, g, a, c, t, ...\n","Shigella flexneri             (a, t, g, a, g, c, g, a, c, c, t, t, g, c, g, ...  (a, t, g, a, g, c, g, a, t, a, t, g, g, c, a, ...\n","Shigella sonnei               (a, t, g, a, g, c, g, a, c, c, t, t, g, c, g, ...  (a, t, g, a, g, c, g, a, t, a, t, g, g, c, a, ...\n","Staphylococcus aureus         (a, t, g, g, c, t, g, a, a, t, t, a, c, c, t, ...  (g, t, g, a, g, t, g, a, a, a, t, a, a, t, t, ...\n","Staphylococcus haemolyticus   (a, t, g, g, c, t, g, a, c, t, t, a, c, c, t, ...  (t, t, g, a, g, t, g, a, g, a, t, a, a, t, t, ...\n","Staphylococcus saprophyticus  (a, t, g, g, c, t, g, a, a, t, t, a, c, c, t, ...  (t, t, g, a, g, t, g, a, a, a, t, a, a, t, t, ...\n","Streptococcus agalactiae      (a, t, g, c, a, a, g, a, t, a, a, a, a, a, t, ...  (a, t, g, t, c, t, a, a, t, a, t, t, c, a, a, ...\n","Streptococcus mitis           (a, t, g, c, a, g, g, a, t, a, g, a, a, a, t, ...  (a, t, g, t, c, t, a, a, c, a, t, t, c, a, a, ...\n","Streptococcus pyogenes        (a, t, g, c, a, a, g, a, t, c, g, a, a, a, t, ...  (a, t, g, t, c, a, a, a, c, a, t, t, c, a, a, ...\n","Streptococcus viridans        (a, t, g, c, a, g, g, a, t, a, a, t, a, a, c, ...  (a, t, g, g, g, a, g, a, g, c, g, c, t, t, t, ..."]},"metadata":{"tags":[]},"execution_count":7}]},{"cell_type":"code","metadata":{"id":"ahf15ytx7tMN","colab":{"base_uri":"https://localhost:8080/","height":1000},"executionInfo":{"status":"ok","timestamp":1623481815209,"user_tz":420,"elapsed":205,"user":{"displayName":"Lila Balakrishnan","photoUrl":"","userId":"05882512769322768995"}},"outputId":"abaa95eb-a5d5-4556-dd2f-2b699ced1aa6"},"source":["df_moxi = pd.merge(df_gyrA_moxi,df_parC_moxi,how=\"inner\", on=['Bacteria Name'],suffixes=(': gyrA moxi',': parC moxi'))\n","df_moxi = df_moxi.sort_index()\n","df_moxi=df_moxi.drop_duplicates(\"Strain: gyrA moxi\",keep='first')\n","df_moxi=df_moxi.drop_duplicates(\"Strain: parC moxi\",keep='first')\n","df_moxi=df_moxi.drop([\"Strain: parC moxi\",'Strain: gyrA moxi'],axis=1)\n","df_moxi.sort_index()"],"execution_count":8,"outputs":[{"output_type":"execute_result","data":{"text/html":["<div>\n","<style scoped>\n","    .dataframe tbody tr th:only-of-type {\n","        vertical-align: middle;\n","    }\n","\n","    .dataframe tbody tr th {\n","        vertical-align: top;\n","    }\n","\n","    .dataframe thead th {\n","        text-align: right;\n","    }\n","</style>\n","<table border=\"1\" class=\"dataframe\">\n","  <thead>\n","    <tr style=\"text-align: right;\">\n","      <th></th>\n","      <th>gyrA Sequence M</th>\n","      <th>parC Sequence M</th>\n","    </tr>\n","    <tr>\n","      <th>Bacteria Name</th>\n","      <th></th>\n","      <th></th>\n","    </tr>\n","  </thead>\n","  <tbody>\n","    <tr>\n","      <th>Bacteroides caccae</th>\n","      <td>(a, t, g, c, t, t, g, a, a, c, a, a, g, a, c, ...</td>\n","      <td>(a, t, g, a, g, t, g, a, c, g, a, a, a, t, t, ...</td>\n","    </tr>\n","    <tr>\n","      <th>Bacteroides fragilis</th>\n","      <td>(a, t, g, c, t, t, g, a, a, c, a, a, g, a, c, ...</td>\n","      <td>(a, t, g, a, g, c, g, a, a, g, a, g, a, a, c, ...</td>\n","    </tr>\n","    <tr>\n","      <th>Bacteroides thetaiotaomicron</th>\n","      <td>(a, t, g, c, t, t, g, a, a, c, a, a, g, a, c, ...</td>\n","      <td>(a, t, g, a, g, t, g, a, c, g, a, a, a, t, c, ...</td>\n","    </tr>\n","    <tr>\n","      <th>Bacteroides vulgatus</th>\n","      <td>(a, t, g, a, t, t, g, a, a, c, t, g, g, g, a, ...</td>\n","      <td>(a, t, g, a, g, c, g, a, c, g, a, t, t, t, t, ...</td>\n","    </tr>\n","    <tr>\n","      <th>Citrobacter freundii</th>\n","      <td>(a, t, g, a, g, c, g, a, c, c, t, t, g, c, g, ...</td>\n","      <td>(a, t, g, a, g, c, g, a, t, a, t, g, g, c, a, ...</td>\n","    </tr>\n","    <tr>\n","      <th>Clostridium perfringens</th>\n","      <td>(a, t, g, a, g, c, t, t, a, a, a, t, g, a, g, ...</td>\n","      <td>(a, t, g, g, c, t, a, a, g, a, a, g, a, a, t, ...</td>\n","    </tr>\n","    <tr>\n","      <th>Enterobacter cloacae</th>\n","      <td>(a, t, g, a, g, c, g, a, c, c, t, t, g, c, g, ...</td>\n","      <td>(a, t, g, a, g, c, g, a, t, a, t, g, g, c, a, ...</td>\n","    </tr>\n","    <tr>\n","      <th>Enterococcus faecalis</th>\n","      <td>(a, t, g, a, g, t, g, a, a, g, a, a, a, t, t, ...</td>\n","      <td>(t, t, g, g, a, a, a, a, a, c, g, c, c, a, a, ...</td>\n","    </tr>\n","    <tr>\n","      <th>Enterococcus faecium</th>\n","      <td>(a, t, g, a, g, t, g, a, a, g, a, a, a, t, c, ...</td>\n","      <td>(a, t, g, g, a, a, a, a, t, c, g, a, c, a, a, ...</td>\n","    </tr>\n","    <tr>\n","      <th>Escherichia coli</th>\n","      <td>(a, t, g, a, g, c, g, a, c, c, t, t, g, c, g, ...</td>\n","      <td>(a, t, g, a, g, c, g, a, t, a, t, g, g, c, a, ...</td>\n","    </tr>\n","    <tr>\n","      <th>Haemophilus influenzae</th>\n","      <td>(a, t, g, a, c, g, g, a, t, t, c, a, a, t, c, ...</td>\n","      <td>(a, t, g, a, c, a, a, a, t, a, t, c, a, a, c, ...</td>\n","    </tr>\n","    <tr>\n","      <th>Haemophilus parainfluenzae</th>\n","      <td>(a, t, g, a, c, g, g, a, t, t, c, a, a, t, c, ...</td>\n","      <td>(a, t, g, g, a, t, c, g, t, g, c, a, t, t, g, ...</td>\n","    </tr>\n","    <tr>\n","      <th>Klebsiella aerogenes</th>\n","      <td>(a, t, g, a, g, c, g, a, c, c, t, t, g, c, g, ...</td>\n","      <td>(a, t, g, a, g, c, g, a, t, a, t, g, g, c, a, ...</td>\n","    </tr>\n","    <tr>\n","      <th>Klebsiella oxytoca</th>\n","      <td>(a, t, g, a, g, c, g, a, c, c, t, t, g, c, g, ...</td>\n","      <td>(a, t, g, a, g, c, g, a, t, a, t, g, g, c, a, ...</td>\n","    </tr>\n","    <tr>\n","      <th>Klebsiella pneumoniae</th>\n","      <td>(a, t, g, a, g, c, g, a, c, c, t, t, g, c, g, ...</td>\n","      <td>(a, t, g, a, g, c, g, a, t, a, t, g, g, c, a, ...</td>\n","    </tr>\n","    <tr>\n","      <th>Listeria monocytogenes</th>\n","      <td>(a, t, g, g, c, a, g, a, a, a, c, a, c, c, a, ...</td>\n","      <td>(t, t, g, a, g, t, a, a, t, c, c, a, g, a, a, ...</td>\n","    </tr>\n","    <tr>\n","      <th>Moraxella catarrhalis</th>\n","      <td>(a, t, g, a, g, t, g, a, t, t, c, g, g, t, c, ...</td>\n","      <td>(a, t, g, a, c, a, a, c, a, c, a, g, a, t, t, ...</td>\n","    </tr>\n","    <tr>\n","      <th>Morganella morganii</th>\n","      <td>(a, t, g, a, g, c, g, a, c, a, a, t, g, c, c, ...</td>\n","      <td>(a, t, g, a, g, t, g, a, a, a, t, a, a, c, t, ...</td>\n","    </tr>\n","    <tr>\n","      <th>Neisseria gonorrhoeae</th>\n","      <td>(a, t, g, a, c, c, g, a, c, g, c, a, a, c, c, ...</td>\n","      <td>(a, t, g, a, a, t, a, c, g, c, a, a, c, c, g, ...</td>\n","    </tr>\n","    <tr>\n","      <th>Proteus mirabilis</th>\n","      <td>(a, t, g, a, g, c, g, a, c, a, t, t, g, c, c, ...</td>\n","      <td>(a, t, g, a, g, t, g, a, a, c, t, g, a, c, t, ...</td>\n","    </tr>\n","    <tr>\n","      <th>Proteus vulgaris</th>\n","      <td>(a, t, g, a, g, c, g, a, c, a, t, t, g, c, c, ...</td>\n","      <td>(a, t, g, a, g, t, g, a, a, c, t, g, a, c, t, ...</td>\n","    </tr>\n","    <tr>\n","      <th>Pseudomonas aeruginosa</th>\n","      <td>(a, t, g, g, g, c, g, a, a, c, t, g, g, c, c, ...</td>\n","      <td>(a, t, g, a, g, c, g, a, a, t, c, c, c, t, c, ...</td>\n","    </tr>\n","    <tr>\n","      <th>Staphylococcus aureus</th>\n","      <td>(a, t, g, g, c, t, g, a, a, t, t, a, c, c, t, ...</td>\n","      <td>(g, t, g, a, g, t, g, a, a, a, t, a, a, t, t, ...</td>\n","    </tr>\n","    <tr>\n","      <th>Staphylococcus epidermidis</th>\n","      <td>(a, t, g, g, c, t, g, a, a, t, t, a, c, c, t, ...</td>\n","      <td>(t, t, g, a, g, t, g, a, g, a, t, a, a, t, t, ...</td>\n","    </tr>\n","    <tr>\n","      <th>Staphylococcus lugdunensis</th>\n","      <td>(a, t, g, g, c, t, g, a, c, t, t, a, c, c, a, ...</td>\n","      <td>(t, t, g, a, g, t, g, a, g, a, t, a, a, t, a, ...</td>\n","    </tr>\n","    <tr>\n","      <th>Staphylococcus saprophyticus</th>\n","      <td>(a, t, g, g, c, t, g, a, a, t, t, a, c, c, t, ...</td>\n","      <td>(t, t, g, a, g, t, g, a, a, a, t, a, a, t, t, ...</td>\n","    </tr>\n","    <tr>\n","      <th>Staphylococcus warneri</th>\n","      <td>(a, t, g, g, c, t, g, a, a, t, t, a, c, c, t, ...</td>\n","      <td>(t, t, g, a, g, t, g, a, g, a, t, a, a, t, t, ...</td>\n","    </tr>\n","    <tr>\n","      <th>Streptococcus agalactiae</th>\n","      <td>(a, t, g, c, a, a, g, a, t, a, a, a, a, a, t, ...</td>\n","      <td>(a, t, g, t, c, t, a, a, t, a, t, t, c, a, a, ...</td>\n","    </tr>\n","    <tr>\n","      <th>Streptococcus intermedius</th>\n","      <td>(a, t, g, c, a, g, g, a, t, a, a, t, a, a, t, ...</td>\n","      <td>(a, t, g, g, g, a, g, a, g, c, g, c, t, t, t, ...</td>\n","    </tr>\n","    <tr>\n","      <th>Streptococcus mitis</th>\n","      <td>(a, t, g, c, a, g, g, a, t, a, g, a, a, a, t, ...</td>\n","      <td>(a, t, g, t, c, t, a, a, c, a, t, t, c, a, a, ...</td>\n","    </tr>\n","    <tr>\n","      <th>Streptococcus pneumoniae</th>\n","      <td>(a, t, g, c, a, g, g, a, t, a, a, a, a, a, t, ...</td>\n","      <td>(a, t, g, g, g, a, g, a, g, c, g, c, t, t, t, ...</td>\n","    </tr>\n","    <tr>\n","      <th>Streptococcus pyogenes</th>\n","      <td>(a, t, g, c, a, a, g, a, t, c, g, a, a, a, t, ...</td>\n","      <td>(a, t, g, t, c, a, a, a, c, a, t, t, c, a, a, ...</td>\n","    </tr>\n","    <tr>\n","      <th>Streptococcus viridans</th>\n","      <td>(a, t, g, c, a, g, g, a, t, a, a, t, a, a, c, ...</td>\n","      <td>(a, t, g, g, g, a, g, a, g, c, g, c, t, t, t, ...</td>\n","    </tr>\n","  </tbody>\n","</table>\n","</div>"],"text/plain":["                                                                gyrA Sequence M                                    parC Sequence M\n","Bacteria Name                                                                                                                     \n","Bacteroides caccae            (a, t, g, c, t, t, g, a, a, c, a, a, g, a, c, ...  (a, t, g, a, g, t, g, a, c, g, a, a, a, t, t, ...\n","Bacteroides fragilis          (a, t, g, c, t, t, g, a, a, c, a, a, g, a, c, ...  (a, t, g, a, g, c, g, a, a, g, a, g, a, a, c, ...\n","Bacteroides thetaiotaomicron  (a, t, g, c, t, t, g, a, a, c, a, a, g, a, c, ...  (a, t, g, a, g, t, g, a, c, g, a, a, a, t, c, ...\n","Bacteroides vulgatus          (a, t, g, a, t, t, g, a, a, c, t, g, g, g, a, ...  (a, t, g, a, g, c, g, a, c, g, a, t, t, t, t, ...\n","Citrobacter freundii          (a, t, g, a, g, c, g, a, c, c, t, t, g, c, g, ...  (a, t, g, a, g, c, g, a, t, a, t, g, g, c, a, ...\n","Clostridium perfringens       (a, t, g, a, g, c, t, t, a, a, a, t, g, a, g, ...  (a, t, g, g, c, t, a, a, g, a, a, g, a, a, t, ...\n","Enterobacter cloacae          (a, t, g, a, g, c, g, a, c, c, t, t, g, c, g, ...  (a, t, g, a, g, c, g, a, t, a, t, g, g, c, a, ...\n","Enterococcus faecalis         (a, t, g, a, g, t, g, a, a, g, a, a, a, t, t, ...  (t, t, g, g, a, a, a, a, a, c, g, c, c, a, a, ...\n","Enterococcus faecium          (a, t, g, a, g, t, g, a, a, g, a, a, a, t, c, ...  (a, t, g, g, a, a, a, a, t, c, g, a, c, a, a, ...\n","Escherichia coli              (a, t, g, a, g, c, g, a, c, c, t, t, g, c, g, ...  (a, t, g, a, g, c, g, a, t, a, t, g, g, c, a, ...\n","Haemophilus influenzae        (a, t, g, a, c, g, g, a, t, t, c, a, a, t, c, ...  (a, t, g, a, c, a, a, a, t, a, t, c, a, a, c, ...\n","Haemophilus parainfluenzae    (a, t, g, a, c, g, g, a, t, t, c, a, a, t, c, ...  (a, t, g, g, a, t, c, g, t, g, c, a, t, t, g, ...\n","Klebsiella aerogenes          (a, t, g, a, g, c, g, a, c, c, t, t, g, c, g, ...  (a, t, g, a, g, c, g, a, t, a, t, g, g, c, a, ...\n","Klebsiella oxytoca            (a, t, g, a, g, c, g, a, c, c, t, t, g, c, g, ...  (a, t, g, a, g, c, g, a, t, a, t, g, g, c, a, ...\n","Klebsiella pneumoniae         (a, t, g, a, g, c, g, a, c, c, t, t, g, c, g, ...  (a, t, g, a, g, c, g, a, t, a, t, g, g, c, a, ...\n","Listeria monocytogenes        (a, t, g, g, c, a, g, a, a, a, c, a, c, c, a, ...  (t, t, g, a, g, t, a, a, t, c, c, a, g, a, a, ...\n","Moraxella catarrhalis         (a, t, g, a, g, t, g, a, t, t, c, g, g, t, c, ...  (a, t, g, a, c, a, a, c, a, c, a, g, a, t, t, ...\n","Morganella morganii           (a, t, g, a, g, c, g, a, c, a, a, t, g, c, c, ...  (a, t, g, a, g, t, g, a, a, a, t, a, a, c, t, ...\n","Neisseria gonorrhoeae         (a, t, g, a, c, c, g, a, c, g, c, a, a, c, c, ...  (a, t, g, a, a, t, a, c, g, c, a, a, c, c, g, ...\n","Proteus mirabilis             (a, t, g, a, g, c, g, a, c, a, t, t, g, c, c, ...  (a, t, g, a, g, t, g, a, a, c, t, g, a, c, t, ...\n","Proteus vulgaris              (a, t, g, a, g, c, g, a, c, a, t, t, g, c, c, ...  (a, t, g, a, g, t, g, a, a, c, t, g, a, c, t, ...\n","Pseudomonas aeruginosa        (a, t, g, g, g, c, g, a, a, c, t, g, g, c, c, ...  (a, t, g, a, g, c, g, a, a, t, c, c, c, t, c, ...\n","Staphylococcus aureus         (a, t, g, g, c, t, g, a, a, t, t, a, c, c, t, ...  (g, t, g, a, g, t, g, a, a, a, t, a, a, t, t, ...\n","Staphylococcus epidermidis    (a, t, g, g, c, t, g, a, a, t, t, a, c, c, t, ...  (t, t, g, a, g, t, g, a, g, a, t, a, a, t, t, ...\n","Staphylococcus lugdunensis    (a, t, g, g, c, t, g, a, c, t, t, a, c, c, a, ...  (t, t, g, a, g, t, g, a, g, a, t, a, a, t, a, ...\n","Staphylococcus saprophyticus  (a, t, g, g, c, t, g, a, a, t, t, a, c, c, t, ...  (t, t, g, a, g, t, g, a, a, a, t, a, a, t, t, ...\n","Staphylococcus warneri        (a, t, g, g, c, t, g, a, a, t, t, a, c, c, t, ...  (t, t, g, a, g, t, g, a, g, a, t, a, a, t, t, ...\n","Streptococcus agalactiae      (a, t, g, c, a, a, g, a, t, a, a, a, a, a, t, ...  (a, t, g, t, c, t, a, a, t, a, t, t, c, a, a, ...\n","Streptococcus intermedius     (a, t, g, c, a, g, g, a, t, a, a, t, a, a, t, ...  (a, t, g, g, g, a, g, a, g, c, g, c, t, t, t, ...\n","Streptococcus mitis           (a, t, g, c, a, g, g, a, t, a, g, a, a, a, t, ...  (a, t, g, t, c, t, a, a, c, a, t, t, c, a, a, ...\n","Streptococcus pneumoniae      (a, t, g, c, a, g, g, a, t, a, a, a, a, a, t, ...  (a, t, g, g, g, a, g, a, g, c, g, c, t, t, t, ...\n","Streptococcus pyogenes        (a, t, g, c, a, a, g, a, t, c, g, a, a, a, t, ...  (a, t, g, t, c, a, a, a, c, a, t, t, c, a, a, ...\n","Streptococcus viridans        (a, t, g, c, a, g, g, a, t, a, a, t, a, a, c, ...  (a, t, g, g, g, a, g, a, g, c, g, c, t, t, t, ..."]},"metadata":{"tags":[]},"execution_count":8}]},{"cell_type":"code","metadata":{"id":"E5JIEGR_730A","executionInfo":{"status":"ok","timestamp":1623481815210,"user_tz":420,"elapsed":203,"user":{"displayName":"Lila Balakrishnan","photoUrl":"","userId":"05882512769322768995"}}},"source":["df = pd.concat([df_moxi,df_cipro],join='outer')\n","# df[['Bacteria Name'],[Bacterioides]]"],"execution_count":9,"outputs":[]},{"cell_type":"markdown","metadata":{"id":"JHLpBEwcX9BD"},"source":["## Make a DataFrame of Kmers"]},{"cell_type":"code","metadata":{"id":"myf2MwSxd0cF","executionInfo":{"status":"ok","timestamp":1623481815211,"user_tz":420,"elapsed":203,"user":{"displayName":"Lila Balakrishnan","photoUrl":"","userId":"05882512769322768995"}}},"source":["#k-mer DF for cipro genes\n","df_1 = makeKmers(df_cipro,'gyrA','C')\n","df_2 = makeKmers(df_cipro,'parC','C')\n","df_words = df_1.merge(df_2,on='Bacteria Name')"],"execution_count":10,"outputs":[]},{"cell_type":"code","metadata":{"id":"Q22JzWHlh3zR","executionInfo":{"status":"ok","timestamp":1623481815212,"user_tz":420,"elapsed":202,"user":{"displayName":"Lila Balakrishnan","photoUrl":"","userId":"05882512769322768995"}}},"source":["#k-mer DF for cipro genes\n","df_1 = makeKmers(df_moxi,'gyrA','M')\n","df_2 = makeKmers(df_moxi,'parC','M')\n","df_words_m = df_1.merge(df_2,on='Bacteria Name')"],"execution_count":11,"outputs":[]},{"cell_type":"code","metadata":{"id":"LtrtWCSyIND_","executionInfo":{"status":"ok","timestamp":1623481815610,"user_tz":420,"elapsed":599,"user":{"displayName":"Lila Balakrishnan","photoUrl":"","userId":"05882512769322768995"}}},"source":["df_words_m.to_csv('/content/drive/Shareddrives/Project2_Drive/PROJECT 2/df_m.csv')"],"execution_count":12,"outputs":[]},{"cell_type":"code","metadata":{"id":"OrhwRqlyx4cW","executionInfo":{"status":"ok","timestamp":1623481815615,"user_tz":420,"elapsed":19,"user":{"displayName":"Lila Balakrishnan","photoUrl":"","userId":"05882512769322768995"}}},"source":["gyrA_C_word = list(df_words['gyrA-C'])\n","for item in range(len(gyrA_C_word)):\n","    gyrA_C_word[item] = ' '.join(gyrA_C_word[item])\n","C_spec = df_words.index.values                         #C_ for C species\n","parC_C_word = list(df_words['parC-C'])\n","for item in range(len(parC_C_word)):\n","    parC_C_word[item] = ' '.join(parC_C_word[item])"],"execution_count":13,"outputs":[]},{"cell_type":"code","metadata":{"id":"3cMyo9bZ0D1E","executionInfo":{"status":"ok","timestamp":1623481815617,"user_tz":420,"elapsed":19,"user":{"displayName":"Lila Balakrishnan","photoUrl":"","userId":"05882512769322768995"}}},"source":["gyrA_M_word = list(df_words_m['gyrA-M'])\n","for item in range(len(gyrA_M_word)):\n","    gyrA_M_word[item] = ' '.join(gyrA_M_word[item])\n","M_spec = df_words_m.index.values                         #M_ for moxi species\n","parC_M_word = list(df_words_m['parC-M'])\n","for item in range(len(parC_M_word)):\n","    parC_M_word[item] = ' '.join(parC_M_word[item])"],"execution_count":14,"outputs":[]},{"cell_type":"code","metadata":{"id":"27a83xGlxcZ-","executionInfo":{"status":"ok","timestamp":1623481815618,"user_tz":420,"elapsed":17,"user":{"displayName":"Lila Balakrishnan","photoUrl":"","userId":"05882512769322768995"}}},"source":["spec=np.append(C_spec,M_spec)\n","gyrA_words=gyrA_C_word+gyrA_M_word\n","parC_words=parC_C_word+parC_M_word\n","cipro_words = gyrA_C_word+parC_C_word"],"execution_count":15,"outputs":[]},{"cell_type":"code","metadata":{"id":"7ttWY_5awWXV","executionInfo":{"status":"ok","timestamp":1623481815620,"user_tz":420,"elapsed":18,"user":{"displayName":"Lila Balakrishnan","photoUrl":"","userId":"05882512769322768995"}}},"source":["df_kmer = pd.concat([df_words,df_words_m],join='outer')"],"execution_count":16,"outputs":[]},{"cell_type":"code","metadata":{"id":"1p2-62zpryE2","executionInfo":{"status":"ok","timestamp":1623481816974,"user_tz":420,"elapsed":1371,"user":{"displayName":"Lila Balakrishnan","photoUrl":"","userId":"05882512769322768995"}}},"source":["from sklearn.feature_extraction.text import CountVectorizer\n","cv = CountVectorizer(ngram_range=(4,4)) #The n-gram size of 4 is previously determined by testing\n","X_gyrA = cv.fit_transform(gyrA_words)\n","X_parC = cv.transform(parC_words)"],"execution_count":17,"outputs":[]},{"cell_type":"markdown","metadata":{"id":"hl2FQdbwnv9-"},"source":["## Make a DataFrame of MIC Values"]},{"cell_type":"code","metadata":{"id":"CmvlLuY54Rhh","executionInfo":{"status":"ok","timestamp":1623481816975,"user_tz":420,"elapsed":251,"user":{"displayName":"Lila Balakrishnan","photoUrl":"","userId":"05882512769322768995"}}},"source":["weighted_MIC = pd.read_csv(\"/content/drive/Shareddrives/Project2_Drive/PROJECT 2/MIC_resistance_all.csv\")\n","weighted_MIC = weighted_MIC.set_index(\"Bacteria Name\")\n","weighted_MIC.dropna(axis=0,how=\"all\",inplace=True)\n","weighted_MIC=weighted_MIC.drop('Staphylococcus hominis',axis=0)"],"execution_count":18,"outputs":[]},{"cell_type":"code","metadata":{"id":"e_Id14TF4Rhi","executionInfo":{"status":"ok","timestamp":1623481816975,"user_tz":420,"elapsed":248,"user":{"displayName":"Lila Balakrishnan","photoUrl":"","userId":"05882512769322768995"}}},"source":["MIC_df_cipro = weighted_MIC[[\"Ciprofloxacin Weighted MIC\"]]\n","MIC_df_cipro = MIC_df_cipro.dropna(how='any')\n","MIC_df_cipro = MIC_df_cipro.rename(columns={\"Ciprofloxacin Weighted MIC\":\"cipro MIC\"})\n","y_cipro = MIC_df_cipro.iloc[:, 0].values # y_cipro for Cipro MIC"],"execution_count":19,"outputs":[]},{"cell_type":"code","metadata":{"id":"iYq9Co45ASEP","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1623481816977,"user_tz":420,"elapsed":249,"user":{"displayName":"Lila Balakrishnan","photoUrl":"","userId":"05882512769322768995"}},"outputId":"1a226a0c-70ff-4c88-d18a-a79d0b0693a7"},"source":["MIC_df_moxi = weighted_MIC[[\"Moxifloxacin Weighted MIC\"]]\n","MIC_df_moxi = MIC_df_moxi.dropna(how='any')\n","MIC_df_moxi = MIC_df_moxi.rename(columns={\"Moxifloxacin Weighted MIC\":\"moxi MIC\"})\n","MIC_df_moxi\n","y_moxi = MIC_df_moxi.iloc[:, 0].values # y_moxi for moxi MIC\n","y_moxi"],"execution_count":20,"outputs":[{"output_type":"execute_result","data":{"text/plain":["array([3.68396226e+00, 2.18169348e+00, 3.05313351e+00, 5.60311189e+00,\n","       2.36377850e+00, 4.28879310e-01, 1.41813830e+00, 1.75797577e+01,\n","       8.09083187e+00, 6.01519416e+00, 3.22200040e-02, 1.49605634e-01,\n","       2.30359903e+00, 1.20064915e+00, 1.51173925e+00, 5.64555838e-01,\n","       7.14902520e-02, 2.01465860e+00, 3.10024390e+00, 1.53618109e+00,\n","       6.19968672e-01, 1.44829224e+01, 8.52192177e-01, 1.33350206e+01,\n","       2.81250000e-01, 2.97509487e+02, 3.12475170e+02, 8.81431325e+01,\n","       1.06800000e-01, 2.96607774e-01, 4.60535685e+00, 2.17774814e+01,\n","       1.86748120e-01])"]},"metadata":{"tags":[]},"execution_count":20}]},{"cell_type":"code","metadata":{"id":"CHYy_q83zAI5","executionInfo":{"status":"ok","timestamp":1623481816992,"user_tz":420,"elapsed":258,"user":{"displayName":"Lila Balakrishnan","photoUrl":"","userId":"05882512769322768995"}}},"source":["y_MIC=np.append(y_cipro,y_moxi)"],"execution_count":21,"outputs":[]},{"cell_type":"code","metadata":{"id":"NS_jz_dJIcSH","colab":{"base_uri":"https://localhost:8080/","height":1000},"executionInfo":{"status":"ok","timestamp":1623481816993,"user_tz":420,"elapsed":54,"user":{"displayName":"Lila Balakrishnan","photoUrl":"","userId":"05882512769322768995"}},"outputId":"8a742721-c2f8-4234-bf6e-2ccec3f352fd"},"source":["MIC_df = pd.concat([MIC_df_cipro,MIC_df_moxi],join='outer')\n","MIC_df"],"execution_count":22,"outputs":[{"output_type":"execute_result","data":{"text/html":["<div>\n","<style scoped>\n","    .dataframe tbody tr th:only-of-type {\n","        vertical-align: middle;\n","    }\n","\n","    .dataframe tbody tr th {\n","        vertical-align: top;\n","    }\n","\n","    .dataframe thead th {\n","        text-align: right;\n","    }\n","</style>\n","<table border=\"1\" class=\"dataframe\">\n","  <thead>\n","    <tr style=\"text-align: right;\">\n","      <th></th>\n","      <th>cipro MIC</th>\n","      <th>moxi MIC</th>\n","    </tr>\n","    <tr>\n","      <th>Bacteria Name</th>\n","      <th></th>\n","      <th></th>\n","    </tr>\n","  </thead>\n","  <tbody>\n","    <tr>\n","      <th>Acinetobacter lwoffii</th>\n","      <td>4.356195</td>\n","      <td>NaN</td>\n","    </tr>\n","    <tr>\n","      <th>Aerococcus urinae</th>\n","      <td>1.133333</td>\n","      <td>NaN</td>\n","    </tr>\n","    <tr>\n","      <th>Alcaligenes faecalis</th>\n","      <td>3.826087</td>\n","      <td>NaN</td>\n","    </tr>\n","    <tr>\n","      <th>Citrobacter braakii</th>\n","      <td>5.358667</td>\n","      <td>NaN</td>\n","    </tr>\n","    <tr>\n","      <th>Citrobacter freundii</th>\n","      <td>1.689000</td>\n","      <td>NaN</td>\n","    </tr>\n","    <tr>\n","      <th>Clostridium perfringens</th>\n","      <td>0.578696</td>\n","      <td>NaN</td>\n","    </tr>\n","    <tr>\n","      <th>Enterobacter cloacae</th>\n","      <td>2.289254</td>\n","      <td>NaN</td>\n","    </tr>\n","    <tr>\n","      <th>Enterococcus faecalis</th>\n","      <td>7.509937</td>\n","      <td>NaN</td>\n","    </tr>\n","    <tr>\n","      <th>Haemophilus influenzae</th>\n","      <td>0.033003</td>\n","      <td>NaN</td>\n","    </tr>\n","    <tr>\n","      <th>Haemophilus parainfluenzae</th>\n","      <td>0.014325</td>\n","      <td>NaN</td>\n","    </tr>\n","    <tr>\n","      <th>Klebsiella oxytoca</th>\n","      <td>1.712428</td>\n","      <td>NaN</td>\n","    </tr>\n","    <tr>\n","      <th>Klebsiella pneumoniae</th>\n","      <td>6.264800</td>\n","      <td>NaN</td>\n","    </tr>\n","    <tr>\n","      <th>Moraxella catarrhalis</th>\n","      <td>0.040766</td>\n","      <td>NaN</td>\n","    </tr>\n","    <tr>\n","      <th>Morganella morganii</th>\n","      <td>3.201296</td>\n","      <td>NaN</td>\n","    </tr>\n","    <tr>\n","      <th>Neisseria gonorrhoeae</th>\n","      <td>5.728179</td>\n","      <td>NaN</td>\n","    </tr>\n","    <tr>\n","      <th>Neisseria meningitidis</th>\n","      <td>0.005864</td>\n","      <td>NaN</td>\n","    </tr>\n","    <tr>\n","      <th>Serratia liquefaciens</th>\n","      <td>4.981385</td>\n","      <td>NaN</td>\n","    </tr>\n","    <tr>\n","      <th>Serratia marcescens</th>\n","      <td>1.851172</td>\n","      <td>NaN</td>\n","    </tr>\n","    <tr>\n","      <th>Shigella flexneri</th>\n","      <td>7.322547</td>\n","      <td>NaN</td>\n","    </tr>\n","    <tr>\n","      <th>Shigella sonnei</th>\n","      <td>3.172596</td>\n","      <td>NaN</td>\n","    </tr>\n","    <tr>\n","      <th>Staphylococcus aureus</th>\n","      <td>4.123794</td>\n","      <td>NaN</td>\n","    </tr>\n","    <tr>\n","      <th>Staphylococcus haemolyticus</th>\n","      <td>32.861450</td>\n","      <td>NaN</td>\n","    </tr>\n","    <tr>\n","      <th>Staphylococcus saprophyticus</th>\n","      <td>0.799645</td>\n","      <td>NaN</td>\n","    </tr>\n","    <tr>\n","      <th>Streptococcus agalactiae</th>\n","      <td>27.479118</td>\n","      <td>NaN</td>\n","    </tr>\n","    <tr>\n","      <th>Streptococcus mitis</th>\n","      <td>2.172205</td>\n","      <td>NaN</td>\n","    </tr>\n","    <tr>\n","      <th>Streptococcus pyogenes</th>\n","      <td>9.894605</td>\n","      <td>NaN</td>\n","    </tr>\n","    <tr>\n","      <th>Streptococcus viridans</th>\n","      <td>5.113930</td>\n","      <td>NaN</td>\n","    </tr>\n","    <tr>\n","      <th>Bacteroides caccae</th>\n","      <td>NaN</td>\n","      <td>3.683962</td>\n","    </tr>\n","    <tr>\n","      <th>Bacteroides fragilis</th>\n","      <td>NaN</td>\n","      <td>2.181693</td>\n","    </tr>\n","    <tr>\n","      <th>Bacteroides thetaiotaomicron</th>\n","      <td>NaN</td>\n","      <td>3.053134</td>\n","    </tr>\n","    <tr>\n","      <th>Bacteroides vulgatus</th>\n","      <td>NaN</td>\n","      <td>5.603112</td>\n","    </tr>\n","    <tr>\n","      <th>Citrobacter freundii</th>\n","      <td>NaN</td>\n","      <td>2.363779</td>\n","    </tr>\n","    <tr>\n","      <th>Clostridium perfringens</th>\n","      <td>NaN</td>\n","      <td>0.428879</td>\n","    </tr>\n","    <tr>\n","      <th>Enterococcus cloacae</th>\n","      <td>NaN</td>\n","      <td>1.418138</td>\n","    </tr>\n","    <tr>\n","      <th>Enterococcus faecalis</th>\n","      <td>NaN</td>\n","      <td>17.579758</td>\n","    </tr>\n","    <tr>\n","      <th>Enterococcus faecium</th>\n","      <td>NaN</td>\n","      <td>8.090832</td>\n","    </tr>\n","    <tr>\n","      <th>Escherichia coli</th>\n","      <td>NaN</td>\n","      <td>6.015194</td>\n","    </tr>\n","    <tr>\n","      <th>Haemophilus influenzae</th>\n","      <td>NaN</td>\n","      <td>0.032220</td>\n","    </tr>\n","    <tr>\n","      <th>Haemophilus parainfluenzae</th>\n","      <td>NaN</td>\n","      <td>0.149606</td>\n","    </tr>\n","    <tr>\n","      <th>Klebsiella aerogenes</th>\n","      <td>NaN</td>\n","      <td>2.303599</td>\n","    </tr>\n","    <tr>\n","      <th>Klebsiella oxytoca</th>\n","      <td>NaN</td>\n","      <td>1.200649</td>\n","    </tr>\n","    <tr>\n","      <th>Klebsiella pneumoniae</th>\n","      <td>NaN</td>\n","      <td>1.511739</td>\n","    </tr>\n","    <tr>\n","      <th>Listeria monocytogenes</th>\n","      <td>NaN</td>\n","      <td>0.564556</td>\n","    </tr>\n","    <tr>\n","      <th>Moraxella catarrhalis</th>\n","      <td>NaN</td>\n","      <td>0.071490</td>\n","    </tr>\n","    <tr>\n","      <th>Morganella morganii</th>\n","      <td>NaN</td>\n","      <td>2.014659</td>\n","    </tr>\n","    <tr>\n","      <th>Neisseria gonorrhoeae</th>\n","      <td>NaN</td>\n","      <td>3.100244</td>\n","    </tr>\n","    <tr>\n","      <th>Proteus mirabilis</th>\n","      <td>NaN</td>\n","      <td>1.536181</td>\n","    </tr>\n","    <tr>\n","      <th>Proteus vulgaris</th>\n","      <td>NaN</td>\n","      <td>0.619969</td>\n","    </tr>\n","    <tr>\n","      <th>Pseudomonas aeruginosa</th>\n","      <td>NaN</td>\n","      <td>14.482922</td>\n","    </tr>\n","    <tr>\n","      <th>Staphylococcus aureus</th>\n","      <td>NaN</td>\n","      <td>0.852192</td>\n","    </tr>\n","    <tr>\n","      <th>Staphylococcus epidermidis</th>\n","      <td>NaN</td>\n","      <td>13.335021</td>\n","    </tr>\n","    <tr>\n","      <th>Staphylococcus lugdunensis</th>\n","      <td>NaN</td>\n","      <td>0.281250</td>\n","    </tr>\n","    <tr>\n","      <th>Staphylococcus saprophyticus</th>\n","      <td>NaN</td>\n","      <td>297.509487</td>\n","    </tr>\n","    <tr>\n","      <th>Staphylococcus warneri</th>\n","      <td>NaN</td>\n","      <td>312.475170</td>\n","    </tr>\n","    <tr>\n","      <th>Streptococcus agalactiae</th>\n","      <td>NaN</td>\n","      <td>88.143133</td>\n","    </tr>\n","    <tr>\n","      <th>Streptococcus intermedius</th>\n","      <td>NaN</td>\n","      <td>0.106800</td>\n","    </tr>\n","    <tr>\n","      <th>Streptococcus mitis</th>\n","      <td>NaN</td>\n","      <td>0.296608</td>\n","    </tr>\n","    <tr>\n","      <th>Streptococcus pneumoniae</th>\n","      <td>NaN</td>\n","      <td>4.605357</td>\n","    </tr>\n","    <tr>\n","      <th>Streptococcus pyogenes</th>\n","      <td>NaN</td>\n","      <td>21.777481</td>\n","    </tr>\n","    <tr>\n","      <th>Streptococcus viridans</th>\n","      <td>NaN</td>\n","      <td>0.186748</td>\n","    </tr>\n","  </tbody>\n","</table>\n","</div>"],"text/plain":["                              cipro MIC    moxi MIC\n","Bacteria Name                                      \n","Acinetobacter lwoffii          4.356195         NaN\n","Aerococcus urinae              1.133333         NaN\n","Alcaligenes faecalis           3.826087         NaN\n","Citrobacter braakii            5.358667         NaN\n","Citrobacter freundii           1.689000         NaN\n","Clostridium perfringens        0.578696         NaN\n","Enterobacter cloacae           2.289254         NaN\n","Enterococcus faecalis          7.509937         NaN\n","Haemophilus influenzae         0.033003         NaN\n","Haemophilus parainfluenzae     0.014325         NaN\n","Klebsiella oxytoca             1.712428         NaN\n","Klebsiella pneumoniae          6.264800         NaN\n","Moraxella catarrhalis          0.040766         NaN\n","Morganella morganii            3.201296         NaN\n","Neisseria gonorrhoeae          5.728179         NaN\n","Neisseria meningitidis         0.005864         NaN\n","Serratia liquefaciens          4.981385         NaN\n","Serratia marcescens            1.851172         NaN\n","Shigella flexneri              7.322547         NaN\n","Shigella sonnei                3.172596         NaN\n","Staphylococcus aureus          4.123794         NaN\n","Staphylococcus haemolyticus   32.861450         NaN\n","Staphylococcus saprophyticus   0.799645         NaN\n","Streptococcus agalactiae      27.479118         NaN\n","Streptococcus mitis            2.172205         NaN\n","Streptococcus pyogenes         9.894605         NaN\n","Streptococcus viridans         5.113930         NaN\n","Bacteroides caccae                  NaN    3.683962\n","Bacteroides fragilis                NaN    2.181693\n","Bacteroides thetaiotaomicron        NaN    3.053134\n","Bacteroides vulgatus                NaN    5.603112\n","Citrobacter freundii                NaN    2.363779\n","Clostridium perfringens             NaN    0.428879\n","Enterococcus cloacae                NaN    1.418138\n","Enterococcus faecalis               NaN   17.579758\n","Enterococcus faecium                NaN    8.090832\n","Escherichia coli                    NaN    6.015194\n","Haemophilus influenzae              NaN    0.032220\n","Haemophilus parainfluenzae          NaN    0.149606\n","Klebsiella aerogenes                NaN    2.303599\n","Klebsiella oxytoca                  NaN    1.200649\n","Klebsiella pneumoniae               NaN    1.511739\n","Listeria monocytogenes              NaN    0.564556\n","Moraxella catarrhalis               NaN    0.071490\n","Morganella morganii                 NaN    2.014659\n","Neisseria gonorrhoeae               NaN    3.100244\n","Proteus mirabilis                   NaN    1.536181\n","Proteus vulgaris                    NaN    0.619969\n","Pseudomonas aeruginosa              NaN   14.482922\n","Staphylococcus aureus               NaN    0.852192\n","Staphylococcus epidermidis          NaN   13.335021\n","Staphylococcus lugdunensis          NaN    0.281250\n","Staphylococcus saprophyticus        NaN  297.509487\n","Staphylococcus warneri              NaN  312.475170\n","Streptococcus agalactiae            NaN   88.143133\n","Streptococcus intermedius           NaN    0.106800\n","Streptococcus mitis                 NaN    0.296608\n","Streptococcus pneumoniae            NaN    4.605357\n","Streptococcus pyogenes              NaN   21.777481\n","Streptococcus viridans              NaN    0.186748"]},"metadata":{"tags":[]},"execution_count":22}]},{"cell_type":"markdown","metadata":{"id":"Ks6IOjoTn8PO"},"source":["## Make a DataFrame of Percent Identity Values"]},{"cell_type":"code","metadata":{"id":"mRQBwXVB4Rhi","executionInfo":{"status":"ok","timestamp":1623481816998,"user_tz":420,"elapsed":43,"user":{"displayName":"Lila Balakrishnan","photoUrl":"","userId":"05882512769322768995"}}},"source":["PIM_df_cipro = pd.read_csv(\"/content/drive/Shareddrives/Project2_Drive/PROJECT 2/ciprofloaxin_PIM_singleval.csv\")\n","PIM_df_cipro = PIM_df_cipro.set_index(\"Bacteria Name\")\n","PIM_df_cipro = PIM_df_cipro.sort_index()\n","PIM_df_cipro = PIM_df_cipro.rename(columns={\"gyrA\":\"c-gyrA\", \"parC\": \"c-parC\"})"],"execution_count":23,"outputs":[]},{"cell_type":"code","metadata":{"id":"zjtie9-BDnad","executionInfo":{"status":"ok","timestamp":1623481817581,"user_tz":420,"elapsed":625,"user":{"displayName":"Lila Balakrishnan","photoUrl":"","userId":"05882512769322768995"}}},"source":["PIM_df_moxi = pd.read_csv(\"/content/drive/Shareddrives/Project2_Drive/PROJECT 2/moxi_PIM.csv\")\n","PIM_df_moxi = PIM_df_moxi.set_index(\"Bacteria Name\")\n","PIM_df_moxi = PIM_df_moxi.sort_index()\n","PIM_df_moxi = PIM_df_moxi.rename(columns={\"gyrA PIM\":\"m-gyrA\", \"parC PIM\": \"m-parC\"})"],"execution_count":24,"outputs":[]},{"cell_type":"code","metadata":{"id":"LIp2V-SIIJib","colab":{"base_uri":"https://localhost:8080/","height":1000},"executionInfo":{"status":"ok","timestamp":1623481817583,"user_tz":420,"elapsed":81,"user":{"displayName":"Lila Balakrishnan","photoUrl":"","userId":"05882512769322768995"}},"outputId":"14c18342-1fe1-40f2-aeba-dcd1a1abbafa"},"source":["PIM_df=pd.concat([PIM_df_cipro,PIM_df_moxi],join='outer',sort=True)\n","PIM_df.groupby(level=0).sum()"],"execution_count":25,"outputs":[{"output_type":"execute_result","data":{"text/html":["<div>\n","<style scoped>\n","    .dataframe tbody tr th:only-of-type {\n","        vertical-align: middle;\n","    }\n","\n","    .dataframe tbody tr th {\n","        vertical-align: top;\n","    }\n","\n","    .dataframe thead th {\n","        text-align: right;\n","    }\n","</style>\n","<table border=\"1\" class=\"dataframe\">\n","  <thead>\n","    <tr style=\"text-align: right;\">\n","      <th></th>\n","      <th>c-gyrA</th>\n","      <th>c-parC</th>\n","      <th>m-gyrA</th>\n","      <th>m-parC</th>\n","    </tr>\n","    <tr>\n","      <th>Bacteria Name</th>\n","      <th></th>\n","      <th></th>\n","      <th></th>\n","      <th></th>\n","    </tr>\n","  </thead>\n","  <tbody>\n","    <tr>\n","      <th>Acinetobacter lwoffii</th>\n","      <td>89.0</td>\n","      <td>93.0</td>\n","      <td>0.0</td>\n","      <td>0.0</td>\n","    </tr>\n","    <tr>\n","      <th>Aerococcus urinae</th>\n","      <td>87.0</td>\n","      <td>94.0</td>\n","      <td>0.0</td>\n","      <td>0.0</td>\n","    </tr>\n","    <tr>\n","      <th>Alcaligenes faecalis</th>\n","      <td>88.0</td>\n","      <td>95.0</td>\n","      <td>0.0</td>\n","      <td>0.0</td>\n","    </tr>\n","    <tr>\n","      <th>Bacteroides caccae</th>\n","      <td>0.0</td>\n","      <td>0.0</td>\n","      <td>96.0</td>\n","      <td>62.0</td>\n","    </tr>\n","    <tr>\n","      <th>Bacteroides fragilis</th>\n","      <td>0.0</td>\n","      <td>0.0</td>\n","      <td>96.0</td>\n","      <td>62.0</td>\n","    </tr>\n","    <tr>\n","      <th>Bacteroides thetaiotaomicron</th>\n","      <td>0.0</td>\n","      <td>0.0</td>\n","      <td>96.0</td>\n","      <td>62.0</td>\n","    </tr>\n","    <tr>\n","      <th>Bacteroides vulgatus</th>\n","      <td>0.0</td>\n","      <td>0.0</td>\n","      <td>97.0</td>\n","      <td>76.0</td>\n","    </tr>\n","    <tr>\n","      <th>Citrobacter braakii</th>\n","      <td>89.0</td>\n","      <td>96.0</td>\n","      <td>0.0</td>\n","      <td>0.0</td>\n","    </tr>\n","    <tr>\n","      <th>Citrobacter freundii</th>\n","      <td>89.0</td>\n","      <td>96.0</td>\n","      <td>96.0</td>\n","      <td>82.0</td>\n","    </tr>\n","    <tr>\n","      <th>Clostridium perfringens</th>\n","      <td>83.0</td>\n","      <td>91.0</td>\n","      <td>97.0</td>\n","      <td>83.0</td>\n","    </tr>\n","    <tr>\n","      <th>Enterobacter cloacae</th>\n","      <td>89.0</td>\n","      <td>94.0</td>\n","      <td>96.0</td>\n","      <td>82.0</td>\n","    </tr>\n","    <tr>\n","      <th>Enterococcus faecalis</th>\n","      <td>87.0</td>\n","      <td>95.0</td>\n","      <td>98.0</td>\n","      <td>86.0</td>\n","    </tr>\n","    <tr>\n","      <th>Enterococcus faecium</th>\n","      <td>0.0</td>\n","      <td>0.0</td>\n","      <td>98.0</td>\n","      <td>86.0</td>\n","    </tr>\n","    <tr>\n","      <th>Escherichia coli</th>\n","      <td>0.0</td>\n","      <td>0.0</td>\n","      <td>96.0</td>\n","      <td>82.0</td>\n","    </tr>\n","    <tr>\n","      <th>Haemophilus influenzae</th>\n","      <td>80.0</td>\n","      <td>89.0</td>\n","      <td>96.0</td>\n","      <td>82.0</td>\n","    </tr>\n","    <tr>\n","      <th>Haemophilus parainfluenzae</th>\n","      <td>88.0</td>\n","      <td>94.0</td>\n","      <td>96.0</td>\n","      <td>82.0</td>\n","    </tr>\n","    <tr>\n","      <th>Klebsiella aerogenes</th>\n","      <td>0.0</td>\n","      <td>0.0</td>\n","      <td>96.0</td>\n","      <td>82.0</td>\n","    </tr>\n","    <tr>\n","      <th>Klebsiella oxytoca</th>\n","      <td>89.0</td>\n","      <td>96.0</td>\n","      <td>96.0</td>\n","      <td>82.0</td>\n","    </tr>\n","    <tr>\n","      <th>Klebsiella pneumoniae</th>\n","      <td>89.0</td>\n","      <td>96.0</td>\n","      <td>96.0</td>\n","      <td>77.0</td>\n","    </tr>\n","    <tr>\n","      <th>Listeria monocytogenes</th>\n","      <td>0.0</td>\n","      <td>0.0</td>\n","      <td>97.0</td>\n","      <td>86.0</td>\n","    </tr>\n","    <tr>\n","      <th>Moraxella catarrhalis</th>\n","      <td>88.0</td>\n","      <td>94.0</td>\n","      <td>96.0</td>\n","      <td>82.0</td>\n","    </tr>\n","    <tr>\n","      <th>Morganella morganii</th>\n","      <td>83.0</td>\n","      <td>89.0</td>\n","      <td>96.0</td>\n","      <td>81.0</td>\n","    </tr>\n","    <tr>\n","      <th>Neisseria gonorrhoeae</th>\n","      <td>87.0</td>\n","      <td>93.0</td>\n","      <td>93.0</td>\n","      <td>79.0</td>\n","    </tr>\n","    <tr>\n","      <th>Neisseria meningitidis</th>\n","      <td>87.0</td>\n","      <td>95.0</td>\n","      <td>0.0</td>\n","      <td>0.0</td>\n","    </tr>\n","    <tr>\n","      <th>Proteus mirabilis</th>\n","      <td>0.0</td>\n","      <td>0.0</td>\n","      <td>96.0</td>\n","      <td>79.0</td>\n","    </tr>\n","    <tr>\n","      <th>Proteus vulgaris</th>\n","      <td>0.0</td>\n","      <td>0.0</td>\n","      <td>96.0</td>\n","      <td>80.0</td>\n","    </tr>\n","    <tr>\n","      <th>Pseudomonas aeruginosa</th>\n","      <td>0.0</td>\n","      <td>0.0</td>\n","      <td>95.0</td>\n","      <td>81.0</td>\n","    </tr>\n","    <tr>\n","      <th>Serratia liquefaciens</th>\n","      <td>89.0</td>\n","      <td>96.0</td>\n","      <td>0.0</td>\n","      <td>0.0</td>\n","    </tr>\n","    <tr>\n","      <th>Serratia marcescens</th>\n","      <td>89.0</td>\n","      <td>96.0</td>\n","      <td>0.0</td>\n","      <td>0.0</td>\n","    </tr>\n","    <tr>\n","      <th>Shigella flexneri</th>\n","      <td>89.0</td>\n","      <td>96.0</td>\n","      <td>0.0</td>\n","      <td>0.0</td>\n","    </tr>\n","    <tr>\n","      <th>Shigella sonnei</th>\n","      <td>89.0</td>\n","      <td>96.0</td>\n","      <td>0.0</td>\n","      <td>0.0</td>\n","    </tr>\n","    <tr>\n","      <th>Staphylococcus aureus</th>\n","      <td>88.0</td>\n","      <td>94.0</td>\n","      <td>196.0</td>\n","      <td>174.0</td>\n","    </tr>\n","    <tr>\n","      <th>Staphylococcus epidermidis</th>\n","      <td>0.0</td>\n","      <td>0.0</td>\n","      <td>98.0</td>\n","      <td>87.0</td>\n","    </tr>\n","    <tr>\n","      <th>Staphylococcus haemolyticus</th>\n","      <td>88.0</td>\n","      <td>94.0</td>\n","      <td>0.0</td>\n","      <td>0.0</td>\n","    </tr>\n","    <tr>\n","      <th>Staphylococcus hominis</th>\n","      <td>88.0</td>\n","      <td>94.0</td>\n","      <td>0.0</td>\n","      <td>0.0</td>\n","    </tr>\n","    <tr>\n","      <th>Staphylococcus lugdunensis</th>\n","      <td>0.0</td>\n","      <td>0.0</td>\n","      <td>96.0</td>\n","      <td>76.0</td>\n","    </tr>\n","    <tr>\n","      <th>Staphylococcus saprophyticus</th>\n","      <td>88.0</td>\n","      <td>95.0</td>\n","      <td>98.0</td>\n","      <td>87.0</td>\n","    </tr>\n","    <tr>\n","      <th>Staphylococcus warneri</th>\n","      <td>0.0</td>\n","      <td>0.0</td>\n","      <td>98.0</td>\n","      <td>87.0</td>\n","    </tr>\n","    <tr>\n","      <th>Streptococcus agalactiae</th>\n","      <td>87.0</td>\n","      <td>94.0</td>\n","      <td>98.0</td>\n","      <td>86.0</td>\n","    </tr>\n","    <tr>\n","      <th>Streptococcus intermedius</th>\n","      <td>0.0</td>\n","      <td>0.0</td>\n","      <td>98.0</td>\n","      <td>86.0</td>\n","    </tr>\n","    <tr>\n","      <th>Streptococcus mitis</th>\n","      <td>88.0</td>\n","      <td>95.0</td>\n","      <td>98.0</td>\n","      <td>86.0</td>\n","    </tr>\n","    <tr>\n","      <th>Streptococcus pneumoniae</th>\n","      <td>0.0</td>\n","      <td>0.0</td>\n","      <td>98.0</td>\n","      <td>86.0</td>\n","    </tr>\n","    <tr>\n","      <th>Streptococcus pyogenes</th>\n","      <td>88.0</td>\n","      <td>95.0</td>\n","      <td>97.0</td>\n","      <td>86.0</td>\n","    </tr>\n","    <tr>\n","      <th>Streptococcus viridans</th>\n","      <td>88.0</td>\n","      <td>95.0</td>\n","      <td>98.0</td>\n","      <td>86.0</td>\n","    </tr>\n","  </tbody>\n","</table>\n","</div>"],"text/plain":["                              c-gyrA  c-parC  m-gyrA  m-parC\n","Bacteria Name                                               \n","Acinetobacter lwoffii           89.0    93.0     0.0     0.0\n","Aerococcus urinae               87.0    94.0     0.0     0.0\n","Alcaligenes faecalis            88.0    95.0     0.0     0.0\n","Bacteroides caccae               0.0     0.0    96.0    62.0\n","Bacteroides fragilis             0.0     0.0    96.0    62.0\n","Bacteroides thetaiotaomicron     0.0     0.0    96.0    62.0\n","Bacteroides vulgatus             0.0     0.0    97.0    76.0\n","Citrobacter braakii             89.0    96.0     0.0     0.0\n","Citrobacter freundii            89.0    96.0    96.0    82.0\n","Clostridium perfringens         83.0    91.0    97.0    83.0\n","Enterobacter cloacae            89.0    94.0    96.0    82.0\n","Enterococcus faecalis           87.0    95.0    98.0    86.0\n","Enterococcus faecium             0.0     0.0    98.0    86.0\n","Escherichia coli                 0.0     0.0    96.0    82.0\n","Haemophilus influenzae          80.0    89.0    96.0    82.0\n","Haemophilus parainfluenzae      88.0    94.0    96.0    82.0\n","Klebsiella aerogenes             0.0     0.0    96.0    82.0\n","Klebsiella oxytoca              89.0    96.0    96.0    82.0\n","Klebsiella pneumoniae           89.0    96.0    96.0    77.0\n","Listeria monocytogenes           0.0     0.0    97.0    86.0\n","Moraxella catarrhalis           88.0    94.0    96.0    82.0\n","Morganella morganii             83.0    89.0    96.0    81.0\n","Neisseria gonorrhoeae           87.0    93.0    93.0    79.0\n","Neisseria meningitidis          87.0    95.0     0.0     0.0\n","Proteus mirabilis                0.0     0.0    96.0    79.0\n","Proteus vulgaris                 0.0     0.0    96.0    80.0\n","Pseudomonas aeruginosa           0.0     0.0    95.0    81.0\n","Serratia liquefaciens           89.0    96.0     0.0     0.0\n","Serratia marcescens             89.0    96.0     0.0     0.0\n","Shigella flexneri               89.0    96.0     0.0     0.0\n","Shigella sonnei                 89.0    96.0     0.0     0.0\n","Staphylococcus aureus           88.0    94.0   196.0   174.0\n","Staphylococcus epidermidis       0.0     0.0    98.0    87.0\n","Staphylococcus haemolyticus     88.0    94.0     0.0     0.0\n","Staphylococcus hominis          88.0    94.0     0.0     0.0\n","Staphylococcus lugdunensis       0.0     0.0    96.0    76.0\n","Staphylococcus saprophyticus    88.0    95.0    98.0    87.0\n","Staphylococcus warneri           0.0     0.0    98.0    87.0\n","Streptococcus agalactiae        87.0    94.0    98.0    86.0\n","Streptococcus intermedius        0.0     0.0    98.0    86.0\n","Streptococcus mitis             88.0    95.0    98.0    86.0\n","Streptococcus pneumoniae         0.0     0.0    98.0    86.0\n","Streptococcus pyogenes          88.0    95.0    97.0    86.0\n","Streptococcus viridans          88.0    95.0    98.0    86.0"]},"metadata":{"tags":[]},"execution_count":25}]},{"cell_type":"markdown","metadata":{"id":"zSnBHNgxoF0m"},"source":["## Combine all data into two DataFrames"]},{"cell_type":"code","metadata":{"id":"VJiSnrBg4Rhj","colab":{"base_uri":"https://localhost:8080/","height":917},"executionInfo":{"status":"ok","timestamp":1623481817585,"user_tz":420,"elapsed":80,"user":{"displayName":"Lila Balakrishnan","photoUrl":"","userId":"05882512769322768995"}},"outputId":"f0e47e64-635a-4ae1-897a-a900e62bd736"},"source":["cipro_df = pd.merge(MIC_df_cipro,PIM_df_cipro,how=\"inner\",on=[\"Bacteria Name\"])\n","cipro_df.drop_duplicates()\n","cipro_df2=pd.merge(cipro_df,df_words,how=\"inner\",on=[\"Bacteria Name\"])\n","cipro_df2.drop_duplicates(subset='cipro MIC')\n","cipro_df2 #ALL CIPRO DATA"],"execution_count":26,"outputs":[{"output_type":"execute_result","data":{"text/html":["<div>\n","<style scoped>\n","    .dataframe tbody tr th:only-of-type {\n","        vertical-align: middle;\n","    }\n","\n","    .dataframe tbody tr th {\n","        vertical-align: top;\n","    }\n","\n","    .dataframe thead th {\n","        text-align: right;\n","    }\n","</style>\n","<table border=\"1\" class=\"dataframe\">\n","  <thead>\n","    <tr style=\"text-align: right;\">\n","      <th></th>\n","      <th>cipro MIC</th>\n","      <th>c-gyrA</th>\n","      <th>c-parC</th>\n","      <th>gyrA-C</th>\n","      <th>parC-C</th>\n","    </tr>\n","    <tr>\n","      <th>Bacteria Name</th>\n","      <th></th>\n","      <th></th>\n","      <th></th>\n","      <th></th>\n","      <th></th>\n","    </tr>\n","  </thead>\n","  <tbody>\n","    <tr>\n","      <th>Acinetobacter lwoffii</th>\n","      <td>4.356195</td>\n","      <td>89</td>\n","      <td>93</td>\n","      <td>[atgagc, tgagcg, gagcgt, agcgta, gcgtat, cgtat...</td>\n","      <td>[atgaca, tgacaa, gacaag, acaagc, caagcc, aagcc...</td>\n","    </tr>\n","    <tr>\n","      <th>Aerococcus urinae</th>\n","      <td>1.133333</td>\n","      <td>87</td>\n","      <td>94</td>\n","      <td>[atggtt, tggttg, ggttga, gttgaa, ttgaag, tgaag...</td>\n","      <td>[atggcg, tggcga, ggcgat, gcgatt, cgattg, gattg...</td>\n","    </tr>\n","    <tr>\n","      <th>Alcaligenes faecalis</th>\n","      <td>3.826087</td>\n","      <td>88</td>\n","      <td>95</td>\n","      <td>[atggat, tggatt, ggattc, gattcc, attcct, ttcct...</td>\n","      <td>[atggac, tggaca, ggacag, gacagc, acagca, cagca...</td>\n","    </tr>\n","    <tr>\n","      <th>Citrobacter braakii</th>\n","      <td>5.358667</td>\n","      <td>89</td>\n","      <td>96</td>\n","      <td>[atgagc, tgagcg, gagcga, agcgac, gcgacc, cgacc...</td>\n","      <td>[atgagc, tgagcg, gagcga, agcgat, gcgata, cgata...</td>\n","    </tr>\n","    <tr>\n","      <th>Citrobacter freundii</th>\n","      <td>1.689000</td>\n","      <td>89</td>\n","      <td>96</td>\n","      <td>[atgagc, tgagcg, gagcga, agcgac, gcgacc, cgacc...</td>\n","      <td>[atgagc, tgagcg, gagcga, agcgat, gcgata, cgata...</td>\n","    </tr>\n","    <tr>\n","      <th>Clostridium perfringens</th>\n","      <td>0.578696</td>\n","      <td>83</td>\n","      <td>91</td>\n","      <td>[atggct, tggcta, ggctaa, gctaag, ctaaga, taaga...</td>\n","      <td>[atgagc, tgagct, gagctt, agctta, gcttaa, cttaa...</td>\n","    </tr>\n","    <tr>\n","      <th>Enterobacter cloacae</th>\n","      <td>2.289254</td>\n","      <td>89</td>\n","      <td>94</td>\n","      <td>[atgagc, tgagcg, gagcga, agcgac, gcgacc, cgacc...</td>\n","      <td>[atgagc, tgagcg, gagcga, agcgat, gcgata, cgata...</td>\n","    </tr>\n","    <tr>\n","      <th>Enterococcus faecalis</th>\n","      <td>7.509937</td>\n","      <td>87</td>\n","      <td>95</td>\n","      <td>[atgagt, tgagtg, gagtga, agtgaa, gtgaag, tgaag...</td>\n","      <td>[ttggaa, tggaaa, ggaaaa, gaaaaa, aaaaac, aaaac...</td>\n","    </tr>\n","    <tr>\n","      <th>Haemophilus influenzae</th>\n","      <td>0.033003</td>\n","      <td>80</td>\n","      <td>89</td>\n","      <td>[atgaca, tgacaa, gacaaa, acaaat, caaata, aaata...</td>\n","      <td>[atgacg, tgacgg, gacgga, acggat, cggatt, ggatt...</td>\n","    </tr>\n","    <tr>\n","      <th>Haemophilus parainfluenzae</th>\n","      <td>0.014325</td>\n","      <td>88</td>\n","      <td>94</td>\n","      <td>[atgacg, tgacgg, gacgga, acggat, cggatt, ggatt...</td>\n","      <td>[atgagc, tgagca, gagcaa, agcaat, gcaata, caata...</td>\n","    </tr>\n","    <tr>\n","      <th>Klebsiella oxytoca</th>\n","      <td>1.712428</td>\n","      <td>89</td>\n","      <td>96</td>\n","      <td>[atgagc, tgagcg, gagcga, agcgac, gcgacc, cgacc...</td>\n","      <td>[atgagc, tgagcg, gagcga, agcgat, gcgata, cgata...</td>\n","    </tr>\n","    <tr>\n","      <th>Klebsiella pneumoniae</th>\n","      <td>6.264800</td>\n","      <td>89</td>\n","      <td>96</td>\n","      <td>[atgagc, tgagcg, gagcga, agcgac, gcgacc, cgacc...</td>\n","      <td>[atgagc, tgagcg, gagcga, agcgat, gcgata, cgata...</td>\n","    </tr>\n","    <tr>\n","      <th>Moraxella catarrhalis</th>\n","      <td>0.040766</td>\n","      <td>88</td>\n","      <td>94</td>\n","      <td>[atgagt, tgagtg, gagtga, agtgat, gtgatt, tgatt...</td>\n","      <td>[atgagt, tgagtg, gagtga, agtgat, gtgatt, tgatt...</td>\n","    </tr>\n","    <tr>\n","      <th>Morganella morganii</th>\n","      <td>3.201296</td>\n","      <td>83</td>\n","      <td>89</td>\n","      <td>[atgagt, tgagtg, gagtga, agtgaa, gtgaaa, tgaaa...</td>\n","      <td>[atgagt, tgagtg, gagtga, agtgaa, gtgaaa, tgaaa...</td>\n","    </tr>\n","    <tr>\n","      <th>Neisseria gonorrhoeae</th>\n","      <td>5.728179</td>\n","      <td>87</td>\n","      <td>93</td>\n","      <td>[atgacc, tgaccg, gaccga, accgac, ccgacg, cgacg...</td>\n","      <td>[atgaat, tgaata, gaatac, aatacg, atacgc, tacgc...</td>\n","    </tr>\n","    <tr>\n","      <th>Neisseria meningitidis</th>\n","      <td>0.005864</td>\n","      <td>87</td>\n","      <td>95</td>\n","      <td>[atgacc, tgaccg, gaccga, accgac, ccgacg, cgacg...</td>\n","      <td>[atgaat, tgaata, gaatac, aatacg, atacgc, tacgc...</td>\n","    </tr>\n","    <tr>\n","      <th>Serratia liquefaciens</th>\n","      <td>4.981385</td>\n","      <td>89</td>\n","      <td>96</td>\n","      <td>[atgagc, tgagcg, gagcga, agcgac, gcgacc, cgacc...</td>\n","      <td>[atgagt, tgagtg, gagtga, agtgat, gtgatc, tgatc...</td>\n","    </tr>\n","    <tr>\n","      <th>Serratia marcescens</th>\n","      <td>1.851172</td>\n","      <td>89</td>\n","      <td>96</td>\n","      <td>[atgagc, tgagcg, gagcga, agcgac, gcgacc, cgacc...</td>\n","      <td>[atgagt, tgagtg, gagtga, agtgat, gtgatc, tgatc...</td>\n","    </tr>\n","    <tr>\n","      <th>Shigella flexneri</th>\n","      <td>7.322547</td>\n","      <td>89</td>\n","      <td>96</td>\n","      <td>[atgagc, tgagcg, gagcga, agcgac, gcgacc, cgacc...</td>\n","      <td>[atgagc, tgagcg, gagcga, agcgat, gcgata, cgata...</td>\n","    </tr>\n","    <tr>\n","      <th>Shigella sonnei</th>\n","      <td>3.172596</td>\n","      <td>89</td>\n","      <td>96</td>\n","      <td>[atgagc, tgagcg, gagcga, agcgac, gcgacc, cgacc...</td>\n","      <td>[atgagc, tgagcg, gagcga, agcgat, gcgata, cgata...</td>\n","    </tr>\n","    <tr>\n","      <th>Staphylococcus aureus</th>\n","      <td>4.123794</td>\n","      <td>88</td>\n","      <td>94</td>\n","      <td>[atggct, tggctg, ggctga, gctgaa, ctgaat, tgaat...</td>\n","      <td>[gtgagt, tgagtg, gagtga, agtgaa, gtgaaa, tgaaa...</td>\n","    </tr>\n","    <tr>\n","      <th>Staphylococcus haemolyticus</th>\n","      <td>32.861450</td>\n","      <td>88</td>\n","      <td>94</td>\n","      <td>[atggct, tggctg, ggctga, gctgac, ctgact, tgact...</td>\n","      <td>[ttgagt, tgagtg, gagtga, agtgag, gtgaga, tgaga...</td>\n","    </tr>\n","    <tr>\n","      <th>Staphylococcus saprophyticus</th>\n","      <td>0.799645</td>\n","      <td>88</td>\n","      <td>95</td>\n","      <td>[atggct, tggctg, ggctga, gctgaa, ctgaat, tgaat...</td>\n","      <td>[ttgagt, tgagtg, gagtga, agtgaa, gtgaaa, tgaaa...</td>\n","    </tr>\n","    <tr>\n","      <th>Streptococcus agalactiae</th>\n","      <td>27.479118</td>\n","      <td>87</td>\n","      <td>94</td>\n","      <td>[atgcaa, tgcaag, gcaaga, caagat, aagata, agata...</td>\n","      <td>[atgtct, tgtcta, gtctaa, tctaat, ctaata, taata...</td>\n","    </tr>\n","    <tr>\n","      <th>Streptococcus mitis</th>\n","      <td>2.172205</td>\n","      <td>88</td>\n","      <td>95</td>\n","      <td>[atgcag, tgcagg, gcagga, caggat, aggata, ggata...</td>\n","      <td>[atgtct, tgtcta, gtctaa, tctaac, ctaaca, taaca...</td>\n","    </tr>\n","    <tr>\n","      <th>Streptococcus pyogenes</th>\n","      <td>9.894605</td>\n","      <td>88</td>\n","      <td>95</td>\n","      <td>[atgcaa, tgcaag, gcaaga, caagat, aagatc, agatc...</td>\n","      <td>[atgtca, tgtcaa, gtcaaa, tcaaac, caaaca, aaaca...</td>\n","    </tr>\n","    <tr>\n","      <th>Streptococcus viridans</th>\n","      <td>5.113930</td>\n","      <td>88</td>\n","      <td>95</td>\n","      <td>[atgcag, tgcagg, gcagga, caggat, aggata, ggata...</td>\n","      <td>[atggga, tgggag, gggaga, ggagag, gagagc, agagc...</td>\n","    </tr>\n","  </tbody>\n","</table>\n","</div>"],"text/plain":["                              cipro MIC  ...                                             parC-C\n","Bacteria Name                            ...                                                   \n","Acinetobacter lwoffii          4.356195  ...  [atgaca, tgacaa, gacaag, acaagc, caagcc, aagcc...\n","Aerococcus urinae              1.133333  ...  [atggcg, tggcga, ggcgat, gcgatt, cgattg, gattg...\n","Alcaligenes faecalis           3.826087  ...  [atggac, tggaca, ggacag, gacagc, acagca, cagca...\n","Citrobacter braakii            5.358667  ...  [atgagc, tgagcg, gagcga, agcgat, gcgata, cgata...\n","Citrobacter freundii           1.689000  ...  [atgagc, tgagcg, gagcga, agcgat, gcgata, cgata...\n","Clostridium perfringens        0.578696  ...  [atgagc, tgagct, gagctt, agctta, gcttaa, cttaa...\n","Enterobacter cloacae           2.289254  ...  [atgagc, tgagcg, gagcga, agcgat, gcgata, cgata...\n","Enterococcus faecalis          7.509937  ...  [ttggaa, tggaaa, ggaaaa, gaaaaa, aaaaac, aaaac...\n","Haemophilus influenzae         0.033003  ...  [atgacg, tgacgg, gacgga, acggat, cggatt, ggatt...\n","Haemophilus parainfluenzae     0.014325  ...  [atgagc, tgagca, gagcaa, agcaat, gcaata, caata...\n","Klebsiella oxytoca             1.712428  ...  [atgagc, tgagcg, gagcga, agcgat, gcgata, cgata...\n","Klebsiella pneumoniae          6.264800  ...  [atgagc, tgagcg, gagcga, agcgat, gcgata, cgata...\n","Moraxella catarrhalis          0.040766  ...  [atgagt, tgagtg, gagtga, agtgat, gtgatt, tgatt...\n","Morganella morganii            3.201296  ...  [atgagt, tgagtg, gagtga, agtgaa, gtgaaa, tgaaa...\n","Neisseria gonorrhoeae          5.728179  ...  [atgaat, tgaata, gaatac, aatacg, atacgc, tacgc...\n","Neisseria meningitidis         0.005864  ...  [atgaat, tgaata, gaatac, aatacg, atacgc, tacgc...\n","Serratia liquefaciens          4.981385  ...  [atgagt, tgagtg, gagtga, agtgat, gtgatc, tgatc...\n","Serratia marcescens            1.851172  ...  [atgagt, tgagtg, gagtga, agtgat, gtgatc, tgatc...\n","Shigella flexneri              7.322547  ...  [atgagc, tgagcg, gagcga, agcgat, gcgata, cgata...\n","Shigella sonnei                3.172596  ...  [atgagc, tgagcg, gagcga, agcgat, gcgata, cgata...\n","Staphylococcus aureus          4.123794  ...  [gtgagt, tgagtg, gagtga, agtgaa, gtgaaa, tgaaa...\n","Staphylococcus haemolyticus   32.861450  ...  [ttgagt, tgagtg, gagtga, agtgag, gtgaga, tgaga...\n","Staphylococcus saprophyticus   0.799645  ...  [ttgagt, tgagtg, gagtga, agtgaa, gtgaaa, tgaaa...\n","Streptococcus agalactiae      27.479118  ...  [atgtct, tgtcta, gtctaa, tctaat, ctaata, taata...\n","Streptococcus mitis            2.172205  ...  [atgtct, tgtcta, gtctaa, tctaac, ctaaca, taaca...\n","Streptococcus pyogenes         9.894605  ...  [atgtca, tgtcaa, gtcaaa, tcaaac, caaaca, aaaca...\n","Streptococcus viridans         5.113930  ...  [atggga, tgggag, gggaga, ggagag, gagagc, agagc...\n","\n","[27 rows x 5 columns]"]},"metadata":{"tags":[]},"execution_count":26}]},{"cell_type":"code","metadata":{"id":"ZRZxTBPbEq83","executionInfo":{"status":"ok","timestamp":1623481818089,"user_tz":420,"elapsed":516,"user":{"displayName":"Lila Balakrishnan","photoUrl":"","userId":"05882512769322768995"}}},"source":["moxi_df = pd.merge(MIC_df_moxi,PIM_df_moxi,how=\"inner\",on=[\"Bacteria Name\"])\n","moxi_df.groupby(level=0)\n","moxi_df=moxi_df.drop_duplicates()"],"execution_count":27,"outputs":[]},{"cell_type":"code","metadata":{"id":"gRxZxEYcOI3S","colab":{"base_uri":"https://localhost:8080/","height":1000},"executionInfo":{"status":"ok","timestamp":1623481818090,"user_tz":420,"elapsed":318,"user":{"displayName":"Lila Balakrishnan","photoUrl":"","userId":"05882512769322768995"}},"outputId":"46f88348-5585-4be2-9f16-9afd239ed457"},"source":["moxi_df2 = pd.merge(moxi_df,df_words_m,how=\"inner\",on=[\"Bacteria Name\"])\n","moxi_df2.drop_duplicates(subset='moxi MIC') #ALL MOXI DATA"],"execution_count":28,"outputs":[{"output_type":"execute_result","data":{"text/html":["<div>\n","<style scoped>\n","    .dataframe tbody tr th:only-of-type {\n","        vertical-align: middle;\n","    }\n","\n","    .dataframe tbody tr th {\n","        vertical-align: top;\n","    }\n","\n","    .dataframe thead th {\n","        text-align: right;\n","    }\n","</style>\n","<table border=\"1\" class=\"dataframe\">\n","  <thead>\n","    <tr style=\"text-align: right;\">\n","      <th></th>\n","      <th>moxi MIC</th>\n","      <th>m-gyrA</th>\n","      <th>m-parC</th>\n","      <th>gyrA-M</th>\n","      <th>parC-M</th>\n","    </tr>\n","    <tr>\n","      <th>Bacteria Name</th>\n","      <th></th>\n","      <th></th>\n","      <th></th>\n","      <th></th>\n","      <th></th>\n","    </tr>\n","  </thead>\n","  <tbody>\n","    <tr>\n","      <th>Bacteroides caccae</th>\n","      <td>3.683962</td>\n","      <td>96</td>\n","      <td>62</td>\n","      <td>[atgctt, tgcttg, gcttga, cttgaa, ttgaac, tgaac...</td>\n","      <td>[atgagt, tgagtg, gagtga, agtgac, gtgacg, tgacg...</td>\n","    </tr>\n","    <tr>\n","      <th>Bacteroides fragilis</th>\n","      <td>2.181693</td>\n","      <td>96</td>\n","      <td>62</td>\n","      <td>[atgctt, tgcttg, gcttga, cttgaa, ttgaac, tgaac...</td>\n","      <td>[atgagc, tgagcg, gagcga, agcgaa, gcgaag, cgaag...</td>\n","    </tr>\n","    <tr>\n","      <th>Bacteroides thetaiotaomicron</th>\n","      <td>3.053134</td>\n","      <td>96</td>\n","      <td>62</td>\n","      <td>[atgctt, tgcttg, gcttga, cttgaa, ttgaac, tgaac...</td>\n","      <td>[atgagt, tgagtg, gagtga, agtgac, gtgacg, tgacg...</td>\n","    </tr>\n","    <tr>\n","      <th>Bacteroides vulgatus</th>\n","      <td>5.603112</td>\n","      <td>97</td>\n","      <td>76</td>\n","      <td>[atgatt, tgattg, gattga, attgaa, ttgaac, tgaac...</td>\n","      <td>[atgagc, tgagcg, gagcga, agcgac, gcgacg, cgacg...</td>\n","    </tr>\n","    <tr>\n","      <th>Citrobacter freundii</th>\n","      <td>2.363779</td>\n","      <td>96</td>\n","      <td>82</td>\n","      <td>[atgagc, tgagcg, gagcga, agcgac, gcgacc, cgacc...</td>\n","      <td>[atgagc, tgagcg, gagcga, agcgat, gcgata, cgata...</td>\n","    </tr>\n","    <tr>\n","      <th>Clostridium perfringens</th>\n","      <td>0.428879</td>\n","      <td>97</td>\n","      <td>83</td>\n","      <td>[atgagc, tgagct, gagctt, agctta, gcttaa, cttaa...</td>\n","      <td>[atggct, tggcta, ggctaa, gctaag, ctaaga, taaga...</td>\n","    </tr>\n","    <tr>\n","      <th>Enterococcus faecalis</th>\n","      <td>17.579758</td>\n","      <td>98</td>\n","      <td>86</td>\n","      <td>[atgagt, tgagtg, gagtga, agtgaa, gtgaag, tgaag...</td>\n","      <td>[ttggaa, tggaaa, ggaaaa, gaaaaa, aaaaac, aaaac...</td>\n","    </tr>\n","    <tr>\n","      <th>Enterococcus faecium</th>\n","      <td>8.090832</td>\n","      <td>98</td>\n","      <td>86</td>\n","      <td>[atgagt, tgagtg, gagtga, agtgaa, gtgaag, tgaag...</td>\n","      <td>[atggaa, tggaaa, ggaaaa, gaaaat, aaaatc, aaatc...</td>\n","    </tr>\n","    <tr>\n","      <th>Escherichia coli</th>\n","      <td>6.015194</td>\n","      <td>96</td>\n","      <td>82</td>\n","      <td>[atgagc, tgagcg, gagcga, agcgac, gcgacc, cgacc...</td>\n","      <td>[atgagc, tgagcg, gagcga, agcgat, gcgata, cgata...</td>\n","    </tr>\n","    <tr>\n","      <th>Haemophilus influenzae</th>\n","      <td>0.032220</td>\n","      <td>96</td>\n","      <td>82</td>\n","      <td>[atgacg, tgacgg, gacgga, acggat, cggatt, ggatt...</td>\n","      <td>[atgaca, tgacaa, gacaaa, acaaat, caaata, aaata...</td>\n","    </tr>\n","    <tr>\n","      <th>Haemophilus parainfluenzae</th>\n","      <td>0.149606</td>\n","      <td>96</td>\n","      <td>82</td>\n","      <td>[atgacg, tgacgg, gacgga, acggat, cggatt, ggatt...</td>\n","      <td>[atggat, tggatc, ggatcg, gatcgt, atcgtg, tcgtg...</td>\n","    </tr>\n","    <tr>\n","      <th>Klebsiella aerogenes</th>\n","      <td>2.303599</td>\n","      <td>96</td>\n","      <td>82</td>\n","      <td>[atgagc, tgagcg, gagcga, agcgac, gcgacc, cgacc...</td>\n","      <td>[atgagc, tgagcg, gagcga, agcgat, gcgata, cgata...</td>\n","    </tr>\n","    <tr>\n","      <th>Klebsiella oxytoca</th>\n","      <td>1.200649</td>\n","      <td>96</td>\n","      <td>82</td>\n","      <td>[atgagc, tgagcg, gagcga, agcgac, gcgacc, cgacc...</td>\n","      <td>[atgagc, tgagcg, gagcga, agcgat, gcgata, cgata...</td>\n","    </tr>\n","    <tr>\n","      <th>Klebsiella pneumoniae</th>\n","      <td>1.511739</td>\n","      <td>96</td>\n","      <td>77</td>\n","      <td>[atgagc, tgagcg, gagcga, agcgac, gcgacc, cgacc...</td>\n","      <td>[atgagc, tgagcg, gagcga, agcgat, gcgata, cgata...</td>\n","    </tr>\n","    <tr>\n","      <th>Listeria monocytogenes</th>\n","      <td>0.564556</td>\n","      <td>97</td>\n","      <td>86</td>\n","      <td>[atggca, tggcag, ggcaga, gcagaa, cagaaa, agaaa...</td>\n","      <td>[ttgagt, tgagta, gagtaa, agtaat, gtaatc, taatc...</td>\n","    </tr>\n","    <tr>\n","      <th>Moraxella catarrhalis</th>\n","      <td>0.071490</td>\n","      <td>96</td>\n","      <td>82</td>\n","      <td>[atgagt, tgagtg, gagtga, agtgat, gtgatt, tgatt...</td>\n","      <td>[atgaca, tgacaa, gacaac, acaaca, caacac, aacac...</td>\n","    </tr>\n","    <tr>\n","      <th>Morganella morganii</th>\n","      <td>2.014659</td>\n","      <td>96</td>\n","      <td>81</td>\n","      <td>[atgagc, tgagcg, gagcga, agcgac, gcgaca, cgaca...</td>\n","      <td>[atgagt, tgagtg, gagtga, agtgaa, gtgaaa, tgaaa...</td>\n","    </tr>\n","    <tr>\n","      <th>Neisseria gonorrhoeae</th>\n","      <td>3.100244</td>\n","      <td>93</td>\n","      <td>79</td>\n","      <td>[atgacc, tgaccg, gaccga, accgac, ccgacg, cgacg...</td>\n","      <td>[atgaat, tgaata, gaatac, aatacg, atacgc, tacgc...</td>\n","    </tr>\n","    <tr>\n","      <th>Proteus mirabilis</th>\n","      <td>1.536181</td>\n","      <td>96</td>\n","      <td>79</td>\n","      <td>[atgagc, tgagcg, gagcga, agcgac, gcgaca, cgaca...</td>\n","      <td>[atgagt, tgagtg, gagtga, agtgaa, gtgaac, tgaac...</td>\n","    </tr>\n","    <tr>\n","      <th>Proteus vulgaris</th>\n","      <td>0.619969</td>\n","      <td>96</td>\n","      <td>80</td>\n","      <td>[atgagc, tgagcg, gagcga, agcgac, gcgaca, cgaca...</td>\n","      <td>[atgagt, tgagtg, gagtga, agtgaa, gtgaac, tgaac...</td>\n","    </tr>\n","    <tr>\n","      <th>Pseudomonas aeruginosa</th>\n","      <td>14.482922</td>\n","      <td>95</td>\n","      <td>81</td>\n","      <td>[atgggc, tgggcg, gggcga, ggcgaa, gcgaac, cgaac...</td>\n","      <td>[atgagc, tgagcg, gagcga, agcgaa, gcgaat, cgaat...</td>\n","    </tr>\n","    <tr>\n","      <th>Staphylococcus aureus</th>\n","      <td>0.852192</td>\n","      <td>98</td>\n","      <td>87</td>\n","      <td>[atggct, tggctg, ggctga, gctgaa, ctgaat, tgaat...</td>\n","      <td>[gtgagt, tgagtg, gagtga, agtgaa, gtgaaa, tgaaa...</td>\n","    </tr>\n","    <tr>\n","      <th>Staphylococcus epidermidis</th>\n","      <td>13.335021</td>\n","      <td>98</td>\n","      <td>87</td>\n","      <td>[atggct, tggctg, ggctga, gctgaa, ctgaat, tgaat...</td>\n","      <td>[ttgagt, tgagtg, gagtga, agtgag, gtgaga, tgaga...</td>\n","    </tr>\n","    <tr>\n","      <th>Staphylococcus lugdunensis</th>\n","      <td>0.281250</td>\n","      <td>96</td>\n","      <td>76</td>\n","      <td>[atggct, tggctg, ggctga, gctgac, ctgact, tgact...</td>\n","      <td>[ttgagt, tgagtg, gagtga, agtgag, gtgaga, tgaga...</td>\n","    </tr>\n","    <tr>\n","      <th>Staphylococcus saprophyticus</th>\n","      <td>297.509487</td>\n","      <td>98</td>\n","      <td>87</td>\n","      <td>[atggct, tggctg, ggctga, gctgaa, ctgaat, tgaat...</td>\n","      <td>[ttgagt, tgagtg, gagtga, agtgaa, gtgaaa, tgaaa...</td>\n","    </tr>\n","    <tr>\n","      <th>Staphylococcus warneri</th>\n","      <td>312.475170</td>\n","      <td>98</td>\n","      <td>87</td>\n","      <td>[atggct, tggctg, ggctga, gctgaa, ctgaat, tgaat...</td>\n","      <td>[ttgagt, tgagtg, gagtga, agtgag, gtgaga, tgaga...</td>\n","    </tr>\n","    <tr>\n","      <th>Streptococcus agalactiae</th>\n","      <td>88.143133</td>\n","      <td>98</td>\n","      <td>86</td>\n","      <td>[atgcaa, tgcaag, gcaaga, caagat, aagata, agata...</td>\n","      <td>[atgtct, tgtcta, gtctaa, tctaat, ctaata, taata...</td>\n","    </tr>\n","    <tr>\n","      <th>Streptococcus intermedius</th>\n","      <td>0.106800</td>\n","      <td>98</td>\n","      <td>86</td>\n","      <td>[atgcag, tgcagg, gcagga, caggat, aggata, ggata...</td>\n","      <td>[atggga, tgggag, gggaga, ggagag, gagagc, agagc...</td>\n","    </tr>\n","    <tr>\n","      <th>Streptococcus mitis</th>\n","      <td>0.296608</td>\n","      <td>98</td>\n","      <td>86</td>\n","      <td>[atgcag, tgcagg, gcagga, caggat, aggata, ggata...</td>\n","      <td>[atgtct, tgtcta, gtctaa, tctaac, ctaaca, taaca...</td>\n","    </tr>\n","    <tr>\n","      <th>Streptococcus pneumoniae</th>\n","      <td>4.605357</td>\n","      <td>98</td>\n","      <td>86</td>\n","      <td>[atgcag, tgcagg, gcagga, caggat, aggata, ggata...</td>\n","      <td>[atggga, tgggag, gggaga, ggagag, gagagc, agagc...</td>\n","    </tr>\n","    <tr>\n","      <th>Streptococcus pyogenes</th>\n","      <td>21.777481</td>\n","      <td>97</td>\n","      <td>86</td>\n","      <td>[atgcaa, tgcaag, gcaaga, caagat, aagatc, agatc...</td>\n","      <td>[atgtca, tgtcaa, gtcaaa, tcaaac, caaaca, aaaca...</td>\n","    </tr>\n","    <tr>\n","      <th>Streptococcus viridans</th>\n","      <td>0.186748</td>\n","      <td>98</td>\n","      <td>86</td>\n","      <td>[atgcag, tgcagg, gcagga, caggat, aggata, ggata...</td>\n","      <td>[atggga, tgggag, gggaga, ggagag, gagagc, agagc...</td>\n","    </tr>\n","  </tbody>\n","</table>\n","</div>"],"text/plain":["                                moxi MIC  ...                                             parC-M\n","Bacteria Name                             ...                                                   \n","Bacteroides caccae              3.683962  ...  [atgagt, tgagtg, gagtga, agtgac, gtgacg, tgacg...\n","Bacteroides fragilis            2.181693  ...  [atgagc, tgagcg, gagcga, agcgaa, gcgaag, cgaag...\n","Bacteroides thetaiotaomicron    3.053134  ...  [atgagt, tgagtg, gagtga, agtgac, gtgacg, tgacg...\n","Bacteroides vulgatus            5.603112  ...  [atgagc, tgagcg, gagcga, agcgac, gcgacg, cgacg...\n","Citrobacter freundii            2.363779  ...  [atgagc, tgagcg, gagcga, agcgat, gcgata, cgata...\n","Clostridium perfringens         0.428879  ...  [atggct, tggcta, ggctaa, gctaag, ctaaga, taaga...\n","Enterococcus faecalis          17.579758  ...  [ttggaa, tggaaa, ggaaaa, gaaaaa, aaaaac, aaaac...\n","Enterococcus faecium            8.090832  ...  [atggaa, tggaaa, ggaaaa, gaaaat, aaaatc, aaatc...\n","Escherichia coli                6.015194  ...  [atgagc, tgagcg, gagcga, agcgat, gcgata, cgata...\n","Haemophilus influenzae          0.032220  ...  [atgaca, tgacaa, gacaaa, acaaat, caaata, aaata...\n","Haemophilus parainfluenzae      0.149606  ...  [atggat, tggatc, ggatcg, gatcgt, atcgtg, tcgtg...\n","Klebsiella aerogenes            2.303599  ...  [atgagc, tgagcg, gagcga, agcgat, gcgata, cgata...\n","Klebsiella oxytoca              1.200649  ...  [atgagc, tgagcg, gagcga, agcgat, gcgata, cgata...\n","Klebsiella pneumoniae           1.511739  ...  [atgagc, tgagcg, gagcga, agcgat, gcgata, cgata...\n","Listeria monocytogenes          0.564556  ...  [ttgagt, tgagta, gagtaa, agtaat, gtaatc, taatc...\n","Moraxella catarrhalis           0.071490  ...  [atgaca, tgacaa, gacaac, acaaca, caacac, aacac...\n","Morganella morganii             2.014659  ...  [atgagt, tgagtg, gagtga, agtgaa, gtgaaa, tgaaa...\n","Neisseria gonorrhoeae           3.100244  ...  [atgaat, tgaata, gaatac, aatacg, atacgc, tacgc...\n","Proteus mirabilis               1.536181  ...  [atgagt, tgagtg, gagtga, agtgaa, gtgaac, tgaac...\n","Proteus vulgaris                0.619969  ...  [atgagt, tgagtg, gagtga, agtgaa, gtgaac, tgaac...\n","Pseudomonas aeruginosa         14.482922  ...  [atgagc, tgagcg, gagcga, agcgaa, gcgaat, cgaat...\n","Staphylococcus aureus           0.852192  ...  [gtgagt, tgagtg, gagtga, agtgaa, gtgaaa, tgaaa...\n","Staphylococcus epidermidis     13.335021  ...  [ttgagt, tgagtg, gagtga, agtgag, gtgaga, tgaga...\n","Staphylococcus lugdunensis      0.281250  ...  [ttgagt, tgagtg, gagtga, agtgag, gtgaga, tgaga...\n","Staphylococcus saprophyticus  297.509487  ...  [ttgagt, tgagtg, gagtga, agtgaa, gtgaaa, tgaaa...\n","Staphylococcus warneri        312.475170  ...  [ttgagt, tgagtg, gagtga, agtgag, gtgaga, tgaga...\n","Streptococcus agalactiae       88.143133  ...  [atgtct, tgtcta, gtctaa, tctaat, ctaata, taata...\n","Streptococcus intermedius       0.106800  ...  [atggga, tgggag, gggaga, ggagag, gagagc, agagc...\n","Streptococcus mitis             0.296608  ...  [atgtct, tgtcta, gtctaa, tctaac, ctaaca, taaca...\n","Streptococcus pneumoniae        4.605357  ...  [atggga, tgggag, gggaga, ggagag, gagagc, agagc...\n","Streptococcus pyogenes         21.777481  ...  [atgtca, tgtcaa, gtcaaa, tcaaac, caaaca, aaaca...\n","Streptococcus viridans          0.186748  ...  [atggga, tgggag, gggaga, ggagag, gagagc, agagc...\n","\n","[32 rows x 5 columns]"]},"metadata":{"tags":[]},"execution_count":28}]},{"cell_type":"markdown","metadata":{"id":"SVGr5kdsoUJc"},"source":["## Data Separation"]},{"cell_type":"code","metadata":{"collapsed":true,"jupyter":{"outputs_hidden":true},"id":"wDXkmNLS4Rhp","executionInfo":{"status":"ok","timestamp":1623481818092,"user_tz":420,"elapsed":312,"user":{"displayName":"Lila Balakrishnan","photoUrl":"","userId":"05882512769322768995"}}},"source":["MIC_val_cipro = np.array(cipro_df['cipro MIC'])\n","columns_list = list(cipro_df.columns)\n","species_list = list(cipro_df.index)\n","cipro_df1 = cipro_df.drop(\"cipro MIC\", axis=1)"],"execution_count":29,"outputs":[]},{"cell_type":"code","metadata":{"id":"p3ydFv2nQ0QQ","executionInfo":{"status":"ok","timestamp":1623481818093,"user_tz":420,"elapsed":310,"user":{"displayName":"Lila Balakrishnan","photoUrl":"","userId":"05882512769322768995"}}},"source":["cipro_df1 = np.array(cipro_df1)"],"execution_count":30,"outputs":[]},{"cell_type":"code","metadata":{"id":"_cebp_J7vEuj","executionInfo":{"status":"ok","timestamp":1623481818093,"user_tz":420,"elapsed":309,"user":{"displayName":"Lila Balakrishnan","photoUrl":"","userId":"05882512769322768995"}}},"source":["MIC_val_moxi = np.array(moxi_df['moxi MIC'])\n","columns_list1 = list(moxi_df.columns)\n","species_list1 = list(moxi_df.index)\n","moxi_df1 = moxi_df.drop(\"moxi MIC\", axis=1)"],"execution_count":31,"outputs":[]},{"cell_type":"code","metadata":{"id":"GxoSh9hIii_d","executionInfo":{"status":"ok","timestamp":1623481818095,"user_tz":420,"elapsed":309,"user":{"displayName":"Lila Balakrishnan","photoUrl":"","userId":"05882512769322768995"}}},"source":["moxi_df1 = np.array(moxi_df1)"],"execution_count":32,"outputs":[]},{"cell_type":"markdown","metadata":{"id":"JS50MSIwoZSR"},"source":["# Random Forest Regression using cipro MIC"]},{"cell_type":"code","metadata":{"id":"kMi9-NzR4Rhr","executionInfo":{"status":"ok","timestamp":1623481818096,"user_tz":420,"elapsed":309,"user":{"displayName":"Lila Balakrishnan","photoUrl":"","userId":"05882512769322768995"}}},"source":["train_features, test_features, train_labels, test_labels = sklearn.model_selection.train_test_split(cipro_df, MIC_val_cipro, test_size = 0.4,random_state=38,shuffle=False)"],"execution_count":33,"outputs":[]},{"cell_type":"code","metadata":{"id":"ZpElVU-P4Rhs","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1623481818096,"user_tz":420,"elapsed":308,"user":{"displayName":"Lila Balakrishnan","photoUrl":"","userId":"05882512769322768995"}},"outputId":"999782c6-d257-4bfc-aead-b9e541fef567"},"source":["print('Training Features Shape:', train_features.shape)\n","print('Training Labels Shape:', train_labels.shape)\n","print('Testing Features Shape:', test_features.shape)\n","print('Testing Labels Shape:', test_labels.shape)"],"execution_count":34,"outputs":[{"output_type":"stream","text":["Training Features Shape: (16, 3)\n","Training Labels Shape: (16,)\n","Testing Features Shape: (11, 3)\n","Testing Labels Shape: (11,)\n"],"name":"stdout"}]},{"cell_type":"code","metadata":{"id":"kCMe3P8x4Rht","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1623481819332,"user_tz":420,"elapsed":1375,"user":{"displayName":"Lila Balakrishnan","photoUrl":"","userId":"05882512769322768995"}},"outputId":"d7c2bb1e-87ca-4519-c31a-f71470998968"},"source":["from sklearn.ensemble import RandomForestRegressor\n","\n","# Instantiate model with 1000 decision trees\n","rf = RandomForestRegressor(n_estimators = 1000, random_state = 18,warm_start=True)\n","\n","# Train the model on training data\n","rf.fit(train_features, train_labels)"],"execution_count":35,"outputs":[{"output_type":"execute_result","data":{"text/plain":["RandomForestRegressor(bootstrap=True, ccp_alpha=0.0, criterion='mse',\n","                      max_depth=None, max_features='auto', max_leaf_nodes=None,\n","                      max_samples=None, min_impurity_decrease=0.0,\n","                      min_impurity_split=None, min_samples_leaf=1,\n","                      min_samples_split=2, min_weight_fraction_leaf=0.0,\n","                      n_estimators=1000, n_jobs=None, oob_score=False,\n","                      random_state=18, verbose=0, warm_start=True)"]},"metadata":{"tags":[]},"execution_count":35}]},{"cell_type":"code","metadata":{"id":"V4NoP_7n4Rhu","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1623481819334,"user_tz":420,"elapsed":332,"user":{"displayName":"Lila Balakrishnan","photoUrl":"","userId":"05882512769322768995"}},"outputId":"bc46c0a6-e0c3-4438-f789-41604109bf80"},"source":["predictions = rf.predict(test_features)\n","\n","# Calculate the absolute errors\n","errors = abs(predictions - test_labels)\n","print(predictions)\n","print(test_labels)\n","\n","# Print out the mean absolute error (mae)\n","print('Mean Absolute Error:', round(np.mean(errors), 2))"],"execution_count":36,"outputs":[{"output_type":"stream","text":["[5.17279808 1.73286418 6.81096128 3.44742671 4.0973224  6.73872015\n"," 0.62853459 6.78013271 1.76458131 6.84113125 5.46721843]\n","[ 4.98138461  1.85117178  7.32254717  3.17259649  4.12379357 32.86144977\n","  0.79964474 27.47911776  2.17220482  9.89460455  5.11393   ]\n","Mean Absolute Error: 4.72\n"],"name":"stdout"}]},{"cell_type":"code","metadata":{"id":"yLBgEti94Rhx","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1623481819336,"user_tz":420,"elapsed":290,"user":{"displayName":"Lila Balakrishnan","photoUrl":"","userId":"05882512769322768995"}},"outputId":"8804489f-290e-4777-a1c2-a87ca2e7e778"},"source":["# Calculate mean absolute percentage error (MAPE)\n","mape = 100 * (errors / test_labels)\n","\n","# Calculate and display accuracy\n","accuracy = 100 - np.mean(mape)\n","print('Accuracy:', round(accuracy, 2), '%.')"],"execution_count":37,"outputs":[{"output_type":"stream","text":["Accuracy: 76.43 %.\n"],"name":"stdout"}]},{"cell_type":"markdown","metadata":{"id":"ECThLa0a2ZfM"},"source":["# Random Forest Regression using Moxi k-mer"]},{"cell_type":"code","metadata":{"id":"rxQSpcY_4Rhy","executionInfo":{"status":"ok","timestamp":1623481819702,"user_tz":420,"elapsed":378,"user":{"displayName":"Lila Balakrishnan","photoUrl":"","userId":"05882512769322768995"}}},"source":["train_featuresm, test_featuresm, train_labelsm, test_labelsm = sklearn.model_selection.train_test_split(moxi_df, MIC_val_moxi, test_size = 0.4,random_state=38,shuffle=False)"],"execution_count":38,"outputs":[]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"enCv7DVxjwOB","executionInfo":{"status":"ok","timestamp":1623481821608,"user_tz":420,"elapsed":2002,"user":{"displayName":"Lila Balakrishnan","photoUrl":"","userId":"05882512769322768995"}},"outputId":"b0f2be91-d64f-4c7c-e7af-580fbeb7f906"},"source":["print('Training Features Shape:', train_featuresm.shape)\n","print('Training Labels Shape:', train_labelsm.shape)\n","print('Testing Features Shape:', test_featuresm.shape)\n","print('Testing Labels Shape:', test_labelsm.shape)\n","from sklearn.ensemble import RandomForestRegressor\n","# Instantiate model with 1000 decision trees\n","rf = RandomForestRegressor(n_estimators = 2000, random_state = 58,warm_start=True)\n","# Train the model on training data\n","rf.fit(train_featuresm, train_labelsm)\n","predictionsm = rf.predict(test_featuresm)\n","# Calculate the absolute errors\n","errorsm = abs(predictionsm - test_labelsm)\n","print(predictionsm)\n","print(test_labelsm)\n","# Print out the mean absolute error (mae)\n","print('Mean Absolute Error:', round(np.mean(errorsm), 2))\n","# Calculate mean absolute percentage error (MAPE)\n","mapem = 100 * (errorsm / test_labelsm)\n","# Calculate and display accuracy\n","accuracym = 100 - np.mean(mapem)\n","print('Accuracy:', round(accuracym, 2), '%.')"],"execution_count":39,"outputs":[{"output_type":"stream","text":["Training Features Shape: (19, 3)\n","Training Labels Shape: (19,)\n","Testing Features Shape: (13, 3)\n","Testing Labels Shape: (13,)\n","[ 0.37283994 12.02669702  2.80493733 13.96822453  0.29964925 13.96822453\n"," 13.96822453 13.96822453  2.46291117  2.53998392  6.27414088 12.460764\n","  2.47587832]\n","[6.19968672e-01 1.44829224e+01 8.52192177e-01 1.33350206e+01\n"," 2.81250000e-01 2.97509487e+02 3.12475170e+02 8.81431325e+01\n"," 1.06800000e-01 2.96607774e-01 4.60535685e+00 2.17774814e+01\n"," 1.86748120e-01]\n","Mean Absolute Error: 52.26\n","Accuracy: -272.27 %.\n"],"name":"stdout"}]},{"cell_type":"markdown","metadata":{"id":"58_GGw-AgCx2"},"source":["# Support Vector Regression"]},{"cell_type":"code","metadata":{"id":"lJS-0XTe4Rhz","executionInfo":{"status":"ok","timestamp":1623481821611,"user_tz":420,"elapsed":10,"user":{"displayName":"Lila Balakrishnan","photoUrl":"","userId":"05882512769322768995"}}},"source":["from sklearn.model_selection import train_test_split\n","from sklearn.svm import SVR\n","from sklearn.metrics import r2_score,mean_squared_error, mean_absolute_error\n","import seaborn as sns\n","import matplotlib.pyplot as plt\n","from sklearn.pipeline import make_pipeline\n","from sklearn.preprocessing import StandardScaler"],"execution_count":40,"outputs":[]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"nJDmqZ89lrC2","executionInfo":{"status":"ok","timestamp":1623481823876,"user_tz":420,"elapsed":2106,"user":{"displayName":"Lila Balakrishnan","photoUrl":"","userId":"05882512769322768995"}},"outputId":"3b0fa96c-8e84-4c87-c4d5-825cc9b0d50d"},"source":["from sklearn.feature_extraction.text import CountVectorizer\n","cv = CountVectorizer(ngram_range=(4,4)) #The n-gram size of 4 is previously determined by testing\n","X_gyrAC = cv.fit_transform(gyrA_C_word)\n","X_parCC = cv.transform(parC_C_word)\n","X_gyrAM = cv.fit_transform(gyrA_M_word)\n","X_parCM = cv.transform(parC_M_word)\n","X_cipro = cv.fit_transform(cipro_words)\n","print(X_gyrAC.shape)\n","print(X_parCC.shape)\n","print(X_gyrAM.shape)\n","print(X_parCM.shape)\n","print(X_cipro.shape)"],"execution_count":41,"outputs":[{"output_type":"stream","text":["(27, 42740)\n","(27, 42740)\n","(33, 48975)\n","(33, 48975)\n","(54, 68857)\n"],"name":"stdout"}]},{"cell_type":"code","metadata":{"id":"zSTx_73TmtDD","executionInfo":{"status":"ok","timestamp":1623481823878,"user_tz":420,"elapsed":64,"user":{"displayName":"Lila Balakrishnan","photoUrl":"","userId":"05882512769322768995"}}},"source":["X_train, X_test, Y_train, Y_test = sklearn.model_selection.train_test_split(X_gyrAC, y_cipro, test_size = 0.4,random_state=38,shuffle=False)"],"execution_count":42,"outputs":[]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"47s7qM9FpVXz","executionInfo":{"status":"ok","timestamp":1623481873180,"user_tz":420,"elapsed":170,"user":{"displayName":"Lila Balakrishnan","photoUrl":"","userId":"05882512769322768995"}},"outputId":"5c74c9fd-24a1-4bb4-c68f-349224a858ce"},"source":["SupportVectorRegModel = SVR()\n","SupportVectorRegModel.fit(X_train,Y_train)\n","Y_pred = SupportVectorRegModel.predict(X_test)\n","print(Y_pred)\n","print(Y_test)\n","mae = mean_absolute_error(Y_test,Y_pred,multioutput='raw_values')\n","er=abs(Y_pred-Y_test)\n","print(mae)\n","perr = 100 * (er/Y_test)\n","print('Accuracy: ', 100- np.mean(perr), '%') #CIPRO PERCENT ACCURACY"],"execution_count":50,"outputs":[{"output_type":"stream","text":["[2.08576253 2.11395013 2.12801082 2.13672888 2.00762582 2.01230012\n"," 2.00683144 2.01352654 2.01385161 2.01380248 2.02696643]\n","[ 4.98138461  1.85117178  7.32254717  3.17259649  4.12379357 32.86144977\n","  0.79964474 27.47911776  2.17220482  9.89460455  5.11393   ]\n","[7.28663805]\n","Accuracy:  35.26862549965166 %\n"],"name":"stdout"}]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"AHwBFF3qvxT6","executionInfo":{"status":"ok","timestamp":1623481823879,"user_tz":420,"elapsed":61,"user":{"displayName":"Lila Balakrishnan","photoUrl":"","userId":"05882512769322768995"}},"outputId":"1dbbd62d-31a6-4d64-d412-965486262558"},"source":["y_cipro2 = np.concatenate((y_cipro,y_cipro))\n","len(y_cipro2)"],"execution_count":44,"outputs":[{"output_type":"execute_result","data":{"text/plain":["54"]},"metadata":{"tags":[]},"execution_count":44}]},{"cell_type":"code","metadata":{"id":"kurOoSN9u4Cr","executionInfo":{"status":"ok","timestamp":1623481823879,"user_tz":420,"elapsed":43,"user":{"displayName":"Lila Balakrishnan","photoUrl":"","userId":"05882512769322768995"}}},"source":["X_train2, X_test2, Y_train2, Y_test2 = sklearn.model_selection.train_test_split(X_cipro, y_cipro2, test_size = 0.4,random_state=38,shuffle=False)"],"execution_count":45,"outputs":[]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"zd3VPNuvwIqI","executionInfo":{"status":"ok","timestamp":1623481823881,"user_tz":420,"elapsed":44,"user":{"displayName":"Lila Balakrishnan","photoUrl":"","userId":"05882512769322768995"}},"outputId":"f774b953-7db3-4f4c-daeb-5f7babd7dab6"},"source":["SupportVectorRegModel = SVR(kernel='rbf')\n","SupportVectorRegModel.fit(X_train2,Y_train2)\n","Y_pred2 = SupportVectorRegModel.predict(X_test2)\n","mae2 = mean_absolute_error(Y_test2,Y_pred2,multioutput='raw_values')\n","er2=abs(Y_pred2-Y_test2)\n","perr2 = 100 * (er2/Y_test2)\n","print(100- np.mean(perr2))"],"execution_count":46,"outputs":[{"output_type":"stream","text":["-4291.011721159459\n"],"name":"stdout"}]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"2_Mhtv5uyVXh","executionInfo":{"status":"ok","timestamp":1623481823881,"user_tz":420,"elapsed":28,"user":{"displayName":"Lila Balakrishnan","photoUrl":"","userId":"05882512769322768995"}},"outputId":"68eae05f-0923-48fa-dffc-a899e219247e"},"source":["X_gyrAC.shape\n","y_cipro.shape"],"execution_count":47,"outputs":[{"output_type":"execute_result","data":{"text/plain":["(27,)"]},"metadata":{"tags":[]},"execution_count":47}]},{"cell_type":"markdown","metadata":{"id":"3aJUraK8WMRe"},"source":["# Model Comparison - Random Forest vs. SVR"]},{"cell_type":"markdown","metadata":{"id":"WeQprIT19TGS"},"source":["This plot is interactive! You can select the models you want to compare using the key on the right."]},{"cell_type":"code","metadata":{"id":"owsPcLjuWLIQ","colab":{"base_uri":"https://localhost:8080/","height":542},"executionInfo":{"status":"ok","timestamp":1623481824988,"user_tz":420,"elapsed":1119,"user":{"displayName":"Lila Balakrishnan","photoUrl":"","userId":"05882512769322768995"}},"outputId":"b32facec-bbea-41d5-8060-04ac4853c102"},"source":["# Imports\n","import plotly.graph_objects as go\n","\n","# Grab the data\n","bacteria_names = ['Species 1', 'Species 2', 'Species 3', 'Species 4', 'Species 5', 'Species 6', 'Species 7', 'Species 8', 'Species 9', 'Species 10', 'Species 11']\n","model1_predictions = predictions\n","model2_predictions = Y_pred\n","actual_values = test_labels\n","\n","# Plot the chart\n","fig = go.Figure(\n","  data=[\n","    go.Bar(name='Random Forest Predicted MIC', x=bacteria_names, y=model1_predictions),\n","    go.Bar(name='SVR Predicted MIC', x=bacteria_names, y=model2_predictions),\n","    # More models could go here\n","    go.Bar(name='Actual MIC', x=bacteria_names, y=actual_values)\n","  ],\n","  layout={\n","    'yaxis': {'title': 'Predicted MIC Value'},\n","    'xaxis': {'title': 'Test Species'},\n","    'title': 'Comparing Predicted MIC Values'\n","  }\n",")\n","fig.update_layout(barmode='group', xaxis_tickangle=-45)\n","fig.show()"],"execution_count":48,"outputs":[{"output_type":"display_data","data":{"text/html":["<html>\n","<head><meta charset=\"utf-8\" /></head>\n","<body>\n","    <div>\n","            <script src=\"https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js?config=TeX-AMS-MML_SVG\"></script><script type=\"text/javascript\">if (window.MathJax) {MathJax.Hub.Config({SVG: {font: \"STIX-Web\"}});}</script>\n","                <script type=\"text/javascript\">window.PlotlyConfig = {MathJaxConfig: 'local'};</script>\n","        <script src=\"https://cdn.plot.ly/plotly-latest.min.js\"></script>    \n","            <div id=\"39a02547-5511-4cfb-8a70-b8ef0113a232\" class=\"plotly-graph-div\" style=\"height:525px; width:100%;\"></div>\n","            <script type=\"text/javascript\">\n","                \n","                    window.PLOTLYENV=window.PLOTLYENV || {};\n","                    \n","                if (document.getElementById(\"39a02547-5511-4cfb-8a70-b8ef0113a232\")) {\n","                    Plotly.newPlot(\n","                        '39a02547-5511-4cfb-8a70-b8ef0113a232',\n","                        [{\"name\": \"Random Forest Predicted MIC\", \"type\": \"bar\", \"x\": [\"Species 1\", \"Species 2\", \"Species 3\", \"Species 4\", \"Species 5\", \"Species 6\", \"Species 7\", \"Species 8\", \"Species 9\", \"Species 10\", \"Species 11\"], \"y\": [5.172798075979957, 1.7328641754530014, 6.810961283636033, 3.447426714487981, 4.097322401766943, 6.738720154117052, 0.6285345893749993, 6.780132705521051, 1.764581308687015, 6.841131249645056, 5.467218430415961]}, {\"name\": \"SVR Predicted MIC\", \"type\": \"bar\", \"x\": [\"Species 1\", \"Species 2\", \"Species 3\", \"Species 4\", \"Species 5\", \"Species 6\", \"Species 7\", \"Species 8\", \"Species 9\", \"Species 10\", \"Species 11\"], \"y\": [2.0857625268920432, 2.113950126078763, 2.1280108171474446, 2.136728880635307, 2.007625818206862, 2.0123001188896814, 2.006831436525869, 2.013526540762213, 2.0138516076448947, 2.0138024838186595, 2.0269664334402657]}, {\"name\": \"Actual MIC\", \"type\": \"bar\", \"x\": [\"Species 1\", \"Species 2\", \"Species 3\", \"Species 4\", \"Species 5\", \"Species 6\", \"Species 7\", \"Species 8\", \"Species 9\", \"Species 10\", \"Species 11\"], \"y\": [4.981384615, 1.851171779, 7.32254717, 3.172596491, 4.123793575, 32.86144977, 0.799644737, 27.47911776, 2.172204819, 9.89460455, 5.11393]}],\n","                        {\"barmode\": \"group\", \"template\": {\"data\": {\"bar\": [{\"error_x\": {\"color\": \"#2a3f5f\"}, \"error_y\": {\"color\": \"#2a3f5f\"}, \"marker\": {\"line\": {\"color\": \"#E5ECF6\", \"width\": 0.5}}, \"type\": \"bar\"}], \"barpolar\": [{\"marker\": {\"line\": {\"color\": \"#E5ECF6\", \"width\": 0.5}}, \"type\": \"barpolar\"}], \"carpet\": [{\"aaxis\": {\"endlinecolor\": \"#2a3f5f\", \"gridcolor\": \"white\", \"linecolor\": \"white\", \"minorgridcolor\": \"white\", \"startlinecolor\": \"#2a3f5f\"}, \"baxis\": {\"endlinecolor\": \"#2a3f5f\", \"gridcolor\": \"white\", \"linecolor\": \"white\", \"minorgridcolor\": \"white\", \"startlinecolor\": \"#2a3f5f\"}, \"type\": \"carpet\"}], \"choropleth\": [{\"colorbar\": {\"outlinewidth\": 0, \"ticks\": \"\"}, \"type\": \"choropleth\"}], \"contour\": [{\"colorbar\": {\"outlinewidth\": 0, \"ticks\": \"\"}, \"colorscale\": [[0.0, \"#0d0887\"], [0.1111111111111111, \"#46039f\"], [0.2222222222222222, \"#7201a8\"], [0.3333333333333333, \"#9c179e\"], [0.4444444444444444, \"#bd3786\"], [0.5555555555555556, \"#d8576b\"], [0.6666666666666666, \"#ed7953\"], [0.7777777777777778, \"#fb9f3a\"], [0.8888888888888888, \"#fdca26\"], [1.0, \"#f0f921\"]], \"type\": \"contour\"}], \"contourcarpet\": [{\"colorbar\": {\"outlinewidth\": 0, \"ticks\": \"\"}, \"type\": \"contourcarpet\"}], \"heatmap\": [{\"colorbar\": {\"outlinewidth\": 0, \"ticks\": \"\"}, \"colorscale\": [[0.0, \"#0d0887\"], [0.1111111111111111, \"#46039f\"], [0.2222222222222222, \"#7201a8\"], [0.3333333333333333, \"#9c179e\"], [0.4444444444444444, \"#bd3786\"], [0.5555555555555556, \"#d8576b\"], [0.6666666666666666, \"#ed7953\"], [0.7777777777777778, \"#fb9f3a\"], [0.8888888888888888, \"#fdca26\"], [1.0, \"#f0f921\"]], \"type\": \"heatmap\"}], \"heatmapgl\": [{\"colorbar\": {\"outlinewidth\": 0, \"ticks\": \"\"}, \"colorscale\": [[0.0, \"#0d0887\"], [0.1111111111111111, \"#46039f\"], [0.2222222222222222, \"#7201a8\"], [0.3333333333333333, \"#9c179e\"], [0.4444444444444444, \"#bd3786\"], [0.5555555555555556, \"#d8576b\"], [0.6666666666666666, \"#ed7953\"], [0.7777777777777778, \"#fb9f3a\"], [0.8888888888888888, \"#fdca26\"], [1.0, \"#f0f921\"]], \"type\": \"heatmapgl\"}], \"histogram\": [{\"marker\": {\"colorbar\": {\"outlinewidth\": 0, \"ticks\": \"\"}}, \"type\": \"histogram\"}], \"histogram2d\": [{\"colorbar\": {\"outlinewidth\": 0, \"ticks\": \"\"}, \"colorscale\": [[0.0, \"#0d0887\"], [0.1111111111111111, \"#46039f\"], [0.2222222222222222, \"#7201a8\"], [0.3333333333333333, \"#9c179e\"], [0.4444444444444444, \"#bd3786\"], [0.5555555555555556, \"#d8576b\"], [0.6666666666666666, \"#ed7953\"], [0.7777777777777778, \"#fb9f3a\"], [0.8888888888888888, \"#fdca26\"], [1.0, \"#f0f921\"]], \"type\": \"histogram2d\"}], \"histogram2dcontour\": [{\"colorbar\": {\"outlinewidth\": 0, \"ticks\": \"\"}, \"colorscale\": [[0.0, \"#0d0887\"], [0.1111111111111111, \"#46039f\"], [0.2222222222222222, \"#7201a8\"], [0.3333333333333333, \"#9c179e\"], [0.4444444444444444, \"#bd3786\"], [0.5555555555555556, \"#d8576b\"], [0.6666666666666666, \"#ed7953\"], [0.7777777777777778, \"#fb9f3a\"], [0.8888888888888888, \"#fdca26\"], [1.0, \"#f0f921\"]], \"type\": \"histogram2dcontour\"}], \"mesh3d\": [{\"colorbar\": {\"outlinewidth\": 0, \"ticks\": \"\"}, \"type\": \"mesh3d\"}], \"parcoords\": [{\"line\": {\"colorbar\": {\"outlinewidth\": 0, \"ticks\": \"\"}}, \"type\": \"parcoords\"}], \"pie\": [{\"automargin\": true, \"type\": \"pie\"}], \"scatter\": [{\"marker\": {\"colorbar\": {\"outlinewidth\": 0, \"ticks\": \"\"}}, \"type\": \"scatter\"}], \"scatter3d\": [{\"line\": {\"colorbar\": {\"outlinewidth\": 0, \"ticks\": \"\"}}, \"marker\": {\"colorbar\": {\"outlinewidth\": 0, \"ticks\": \"\"}}, \"type\": \"scatter3d\"}], \"scattercarpet\": [{\"marker\": {\"colorbar\": {\"outlinewidth\": 0, \"ticks\": \"\"}}, \"type\": \"scattercarpet\"}], \"scattergeo\": [{\"marker\": {\"colorbar\": {\"outlinewidth\": 0, \"ticks\": \"\"}}, \"type\": \"scattergeo\"}], \"scattergl\": [{\"marker\": {\"colorbar\": {\"outlinewidth\": 0, \"ticks\": \"\"}}, \"type\": \"scattergl\"}], \"scattermapbox\": [{\"marker\": {\"colorbar\": {\"outlinewidth\": 0, \"ticks\": \"\"}}, \"type\": \"scattermapbox\"}], \"scatterpolar\": [{\"marker\": {\"colorbar\": {\"outlinewidth\": 0, \"ticks\": \"\"}}, \"type\": \"scatterpolar\"}], \"scatterpolargl\": [{\"marker\": {\"colorbar\": {\"outlinewidth\": 0, \"ticks\": \"\"}}, \"type\": \"scatterpolargl\"}], \"scatterternary\": [{\"marker\": {\"colorbar\": {\"outlinewidth\": 0, \"ticks\": \"\"}}, \"type\": \"scatterternary\"}], \"surface\": [{\"colorbar\": {\"outlinewidth\": 0, \"ticks\": \"\"}, \"colorscale\": [[0.0, \"#0d0887\"], [0.1111111111111111, \"#46039f\"], [0.2222222222222222, \"#7201a8\"], [0.3333333333333333, \"#9c179e\"], [0.4444444444444444, \"#bd3786\"], [0.5555555555555556, \"#d8576b\"], [0.6666666666666666, \"#ed7953\"], [0.7777777777777778, \"#fb9f3a\"], [0.8888888888888888, \"#fdca26\"], [1.0, \"#f0f921\"]], \"type\": \"surface\"}], \"table\": [{\"cells\": {\"fill\": {\"color\": \"#EBF0F8\"}, \"line\": {\"color\": \"white\"}}, \"header\": {\"fill\": {\"color\": \"#C8D4E3\"}, \"line\": {\"color\": \"white\"}}, \"type\": \"table\"}]}, \"layout\": {\"annotationdefaults\": {\"arrowcolor\": \"#2a3f5f\", \"arrowhead\": 0, \"arrowwidth\": 1}, \"coloraxis\": {\"colorbar\": {\"outlinewidth\": 0, \"ticks\": \"\"}}, \"colorscale\": {\"diverging\": [[0, \"#8e0152\"], [0.1, \"#c51b7d\"], [0.2, \"#de77ae\"], [0.3, \"#f1b6da\"], [0.4, \"#fde0ef\"], [0.5, \"#f7f7f7\"], [0.6, \"#e6f5d0\"], [0.7, \"#b8e186\"], [0.8, \"#7fbc41\"], [0.9, \"#4d9221\"], [1, \"#276419\"]], \"sequential\": [[0.0, \"#0d0887\"], [0.1111111111111111, \"#46039f\"], [0.2222222222222222, \"#7201a8\"], [0.3333333333333333, \"#9c179e\"], [0.4444444444444444, \"#bd3786\"], [0.5555555555555556, \"#d8576b\"], [0.6666666666666666, \"#ed7953\"], [0.7777777777777778, \"#fb9f3a\"], [0.8888888888888888, \"#fdca26\"], [1.0, \"#f0f921\"]], \"sequentialminus\": [[0.0, \"#0d0887\"], [0.1111111111111111, \"#46039f\"], [0.2222222222222222, \"#7201a8\"], [0.3333333333333333, \"#9c179e\"], [0.4444444444444444, \"#bd3786\"], [0.5555555555555556, \"#d8576b\"], [0.6666666666666666, \"#ed7953\"], [0.7777777777777778, \"#fb9f3a\"], [0.8888888888888888, \"#fdca26\"], [1.0, \"#f0f921\"]]}, \"colorway\": [\"#636efa\", \"#EF553B\", \"#00cc96\", \"#ab63fa\", \"#FFA15A\", \"#19d3f3\", \"#FF6692\", \"#B6E880\", \"#FF97FF\", \"#FECB52\"], \"font\": {\"color\": \"#2a3f5f\"}, \"geo\": {\"bgcolor\": \"white\", \"lakecolor\": \"white\", \"landcolor\": \"#E5ECF6\", \"showlakes\": true, \"showland\": true, \"subunitcolor\": \"white\"}, \"hoverlabel\": {\"align\": \"left\"}, \"hovermode\": \"closest\", \"mapbox\": {\"style\": \"light\"}, \"paper_bgcolor\": \"white\", \"plot_bgcolor\": \"#E5ECF6\", \"polar\": {\"angularaxis\": {\"gridcolor\": \"white\", \"linecolor\": \"white\", \"ticks\": \"\"}, \"bgcolor\": \"#E5ECF6\", \"radialaxis\": {\"gridcolor\": \"white\", \"linecolor\": \"white\", \"ticks\": \"\"}}, \"scene\": {\"xaxis\": {\"backgroundcolor\": \"#E5ECF6\", \"gridcolor\": \"white\", \"gridwidth\": 2, \"linecolor\": \"white\", \"showbackground\": true, \"ticks\": \"\", \"zerolinecolor\": \"white\"}, \"yaxis\": {\"backgroundcolor\": \"#E5ECF6\", \"gridcolor\": \"white\", \"gridwidth\": 2, \"linecolor\": \"white\", \"showbackground\": true, \"ticks\": \"\", \"zerolinecolor\": \"white\"}, \"zaxis\": {\"backgroundcolor\": \"#E5ECF6\", \"gridcolor\": \"white\", \"gridwidth\": 2, \"linecolor\": \"white\", \"showbackground\": true, \"ticks\": \"\", \"zerolinecolor\": \"white\"}}, \"shapedefaults\": {\"line\": {\"color\": \"#2a3f5f\"}}, \"ternary\": {\"aaxis\": {\"gridcolor\": \"white\", \"linecolor\": \"white\", \"ticks\": \"\"}, \"baxis\": {\"gridcolor\": \"white\", \"linecolor\": \"white\", \"ticks\": \"\"}, \"bgcolor\": \"#E5ECF6\", \"caxis\": {\"gridcolor\": \"white\", \"linecolor\": \"white\", \"ticks\": \"\"}}, \"title\": {\"x\": 0.05}, \"xaxis\": {\"automargin\": true, \"gridcolor\": \"white\", \"linecolor\": \"white\", \"ticks\": \"\", \"title\": {\"standoff\": 15}, \"zerolinecolor\": \"white\", \"zerolinewidth\": 2}, \"yaxis\": {\"automargin\": true, \"gridcolor\": \"white\", \"linecolor\": \"white\", \"ticks\": \"\", \"title\": {\"standoff\": 15}, \"zerolinecolor\": \"white\", \"zerolinewidth\": 2}}}, \"title\": {\"text\": \"Comparing Predicted MIC Values\"}, \"xaxis\": {\"tickangle\": -45, \"title\": {\"text\": \"Test Species\"}}, \"yaxis\": {\"title\": {\"text\": \"Predicted MIC Value\"}}},\n","                        {\"responsive\": true}\n","                    ).then(function(){\n","                            \n","var gd = document.getElementById('39a02547-5511-4cfb-8a70-b8ef0113a232');\n","var x = new MutationObserver(function (mutations, observer) {{\n","        var display = window.getComputedStyle(gd).display;\n","        if (!display || display === 'none') {{\n","            console.log([gd, 'removed!']);\n","            Plotly.purge(gd);\n","            observer.disconnect();\n","        }}\n","}});\n","\n","// Listen for the removal of the full notebook cells\n","var notebookContainer = gd.closest('#notebook-container');\n","if (notebookContainer) {{\n","    x.observe(notebookContainer, {childList: true});\n","}}\n","\n","// Listen for the clearing of the current output cell\n","var outputEl = gd.closest('.output');\n","if (outputEl) {{\n","    x.observe(outputEl, {childList: true});\n","}}\n","\n","                        })\n","                };\n","                \n","            </script>\n","        </div>\n","</body>\n","</html>"]},"metadata":{"tags":[]}}]}]}